\documentclass[]{article}
\usepackage{xcolor}
\usepackage{lineno}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage[margin=1in]{geometry}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdftitle={Shifting spaces: which disparity or dissimilarity measurement best summarise occupancy in multidimensional spaces?},
            pdfauthor={Thomas Guillerme, Mark N. Puttick, Ariel E. Marcy, Vera Weisbecker},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{longtable,booktabs}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\providecommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

  \title{Shifting spaces: which disparity or dissimilarity measurement best
summarise occupancy in multidimensional spaces?}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
    \author{Thomas Guillerme, Mark N. Puttick, Ariel E. Marcy, Vera Weisbecker}
    \preauthor{\centering\large\emph}
  \postauthor{\par}
      \predate{\centering\large\emph}
  \postdate{\par}
    \date{2020-03-02}

\linespread{1.6} %in the document header

\begin{document}
\maketitle
\modulolinenumbers[1] % just after the \begin{document} tag
\linenumbers
\section{Abstract}\label{abstract}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Multidimensional analysis of traits are now a common toolkit in
  ecology and evolution and are based on
  \textcolor{blue}{trait spaces} in which each dimension
  summarises the observed trait combination (a morphospace or an
  ecospace). Observations of interest will typically occupy a
  \textcolor{blue}{subspace of this space, and researchers will calculate one or more measures to quantify the way in which organisms "inhabit" that space.}
  In macroevolution and ecology these measures are referred to as
  disparity or dissimilarity metrics and can be generalised as
  \textcolor{blue}{space occupancy measures (the distribution of the data in space).}
  Researchers use these measures to investigate how space occupancy
  changes through time, in relation to other groups of organisms, and in
  response to global environmental changes. However, the mathematical
  and biological meaning of most space occupancy measures is vague with
  the majority of widely-used measures lacking formal description.
\item
  Here we propose a broad classification of space occupancy measures
  into three categories that capture changes in
  \textcolor{blue}{size}, density, or position. We analyse
  the behaviour of 25 measures to study changes in
  \textcolor{blue}{trait space size}, density and position
  on a series of simulated and empirical datasets.
\item
  We find no one measure describes all of
  \textcolor{blue}{trait space} but that some measures are
  better at capturing certain aspects and that their performance depends
  on both the \textcolor{blue}{trait space} and the
  hypothesis analysed. Our results confirm the three broad categories
  (\textcolor{blue}{size}, density and position) and allow
  us to relate changes in any of these categories to biological
  phenomena.
\item
  Since the choice of space occupancy measures is specific to the data
  and question, we introduced
  \href{https://tguillerme.shinyapps.io/moms/}{\texttt{moms}}, a tool
  based allowing users to both visualise and capture changes in space
  occupancy for any measurement.
  \href{https://tguillerme.shinyapps.io/moms/}{\texttt{moms}} is
  designed to help workers choose the right space occupancy measures,
  given the properties of their
  \textcolor{blue}{trait space} and their biological
  question.
  \textcolor{blue}{By providing guidelines and common vocabulary for space occupancy analysis, we hope to help bridging the gap in multidimensional research between ecology and evolution.}
\end{enumerate}

\section{Introduction}\label{introduction}

Groups of species and environments share specific, recognisable,
correlated characteristics: guilds or biomes with shared phenotypic,
physiological, phylogenetic or behavioural traits. Organisms or
environments should therefore be studied as a set of traits rather than
some specific traits in isolation (Donohue et al. 2013; Hopkins and
Gerber 2017). Biologists have increasingly been using ordination
techniques (see Legendre and Legendre 2012 for a summary) to create
multidimensional \textcolor{blue}{trait spaces} to either
explore properties of the data or test hypotheses (e.g. Oksanen et al.
2007; Blonder 2018; Guillerme 2018). For example, in palaeobiology,
Wright (2017) \textcolor{blue}{used trait spaces} to study
how groups of species' characteristics change through time; in ecology,
Jones et al. (2015) study evidence of competition by looking at trait
overlap between two populations. However, different fields use a
different set of terms for such approaches (Table 1). Nonetheless, they
use the same mathematical objects: matrices with columns representing an
original or transformed trait value and rows representing observations
(taxon, field site, etc.; Guillerme 2018).


\renewcommand\baselinestretch{1}\selectfont

\begin{longtable}[]{llll}
\toprule
\begin{minipage}[b]{0.2111\columnwidth}\raggedright\strut
Mathematics\strut
\end{minipage} & \begin{minipage}[b]{0.2111\columnwidth}\raggedright\strut
Ecology\strut
\end{minipage} & \begin{minipage}[b]{0.2111\columnwidth}\raggedright\strut
Macroevolution\strut
\end{minipage} & \begin{minipage}[b]{0.2111\columnwidth}\raggedright\strut
This paper\strut
\end{minipage}\tabularnewline
\midrule
\endhead
\begin{minipage}[t]{0.2111\columnwidth}\raggedright\strut
Matrix (\(n \times d\))
\textcolor{blue}{with a structural relation between rows and columns}\strut
\end{minipage} & \begin{minipage}[t]{0.2111\columnwidth}\raggedright\strut
Function-space, Eco-space, etc.\strut
\end{minipage} & \begin{minipage}[t]{0.2111\columnwidth}\raggedright\strut
Morphospace, traitspace, etc.\strut
\end{minipage} & \begin{minipage}[t]{0.2111\columnwidth}\raggedright\strut
trait space\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.2111\columnwidth}\raggedright\strut
Rows (\emph{n})\strut
\end{minipage} & \begin{minipage}[t]{0.2111\columnwidth}\raggedright\strut
Taxa, field sites, environments, etc.\strut
\end{minipage} & \begin{minipage}[t]{0.2111\columnwidth}\raggedright\strut
Taxa, specimen, populations, etc.\strut
\end{minipage} & \begin{minipage}[t]{0.2111\columnwidth}\raggedright\strut
observations\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.2111\columnwidth}\raggedright\strut
Columns (\emph{d})\strut
\end{minipage} & \begin{minipage}[t]{0.2111\columnwidth}\raggedright\strut
Traits, Ordination scores, distances, etc.\strut
\end{minipage} & \begin{minipage}[t]{0.2111\columnwidth}\raggedright\strut
Traits, Ordination scores, distances, etc.\strut
\end{minipage} & \begin{minipage}[t]{0.2111\columnwidth}\raggedright\strut
dimensions\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.2111\columnwidth}\raggedright\strut
Matrix subset (\(m \times d\); \(m \leq n\))\strut
\end{minipage} & \begin{minipage}[t]{0.2111\columnwidth}\raggedright\strut
Treatments, phylogenetic group (clade), etc.\strut
\end{minipage} & \begin{minipage}[t]{0.2111\columnwidth}\raggedright\strut
Clades, geological stratum, etc.\strut
\end{minipage} & \begin{minipage}[t]{0.2111\columnwidth}\raggedright\strut
group\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.2111\columnwidth}\raggedright\strut
Statistic\strut
\end{minipage} & \begin{minipage}[t]{0.2111\columnwidth}\raggedright\strut
Dissimilarity index or metric, hypervolume, functional diversity\strut
\end{minipage} & \begin{minipage}[t]{0.2111\columnwidth}\raggedright\strut
Disparity metric or index\strut
\end{minipage} & \begin{minipage}[t]{0.2111\columnwidth}\raggedright\strut
space occupancy measure\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.2111\columnwidth}\raggedright\strut
Multidimensional analysis\strut
\end{minipage} & \begin{minipage}[t]{0.2111\columnwidth}\raggedright\strut
Dissimilarity analysis, trait analysis, etc.\strut
\end{minipage} & \begin{minipage}[t]{0.2111\columnwidth}\raggedright\strut
Disparity analysis, disparity-through-time, etc.\strut
\end{minipage} & \begin{minipage}[t]{0.2111\columnwidth}\raggedright\strut
multidimensional analysis\strut
\end{minipage}\tabularnewline
\bottomrule
\caption{terms and equivalence between mathematics, ecology and
macroevolution.}
\end{longtable}

\renewcommand\baselinestretch{1.6}\selectfont

Ecologists and evolutionary biologists often use
\textcolor{blue}{trait spaces} with respect to the same
fundamental questions:
\textcolor{blue}{are groups occupying the same amount of trait space?
Do some groups contain more species than others in the same amount of trait space?
Are some specific factors correlated with different patterns of trait space occupancy?
Because of the multidimensional nature of these trait spaces, it is often not possible to study them using bi- or tri-variate techniques [@diaz2016; @hopkins2017; @mammola2019].}
Studying the occupancy of
\textcolor{blue}{trait spaces is done using disparity indices}
in macroevolution (Wills 2001; Hopkins and Gerber 2017; Guillerme 2018)
or comparing hypervolumes in ecology (Donohue et al. 2013; Díaz et al.
2016; Blonder 2018; Mammola 2019). Despite the commonalities between
\textcolor{blue}{the measures used in ecology and evolution (which are often metric but don't necessarily need to be)},
surprisingly little work has been published on their behaviour (but see
Ciampaglio et al. 2001; Villéger et al. 2008; Mammola 2019).

Different
\textcolor{blue}{occupancy measures capture different aspects of  trait space}
(ciampaglio2001; Villéger et al. 2008; Mammola 2019). It may be
widely-known, but to our knowledge is infrequently mentioned in
peer-reviewed papers. First,
\textcolor{blue}{space occupancy measures} are often named
as the biological aspect they are describing (``disparity'',
``functional diversity'') rather than what they are measuring (e.g.~the
product of ranges), which obscures the differences and similarities
between studies. Second, in many studies in ecology and evolution,
authors have focused on measuring the
\textcolor{blue}{size of the trait space} (e.g.~ellipsoid
volume Donohue et al. 2013; hypervolume Díaz et al. 2016; Procrustes
variance Marcy et al. 2016; product of variance Wright 2017). However,
\textcolor{blue}{the size of the trait space} only
represents one aspects of occupancy, disregarding others such as the
density (Harmon et al. 2008) or position (Wills 2001; Ciampaglio et al.
2001). For example, if two groups
\textcolor{blue}{have the same size}, this can support
certain biological conclusions. Yet, an alternative aspect of space
occupancy may indicate that the groups' position are different, leading
to a different biological conclusion (e.g.~the groups are equally
diverse but occupy different niches).
\textcolor{blue}{Using measures that only capture one aspect of the trait space}
may restrain the potential of multidimensional analysis (Villéger et al.
2008).

Here we propose a broad classification of space occupancy measures as
used across ecology and evolution and study their power to detect
changes in \textcolor{blue}{trait space} occupancy in
simulated and empirical data
(\textcolor{blue}{regardless of whether spaces are are truly "occupiable" which might be important in some cases - e.g. if the space is infinite or if some regions inapplicable)}.
We provide an assessment of each broad type of space occupancy measures
along with a unified terminology to foster communication between ecology
and evolution. Unsurprisingly, we found no one measure describes all
changes and that the results from each measures are dependent on the
characteristics of the space and the hypotheses. Furthermore, because
there can be an infinite number of measures, it would be impossible to
propose clear generalities to space occupancy measures behaviour.
Therefore, we propose
\href{https://tguillerme.shinyapps.io/moms/}{\texttt{moms}}, a tool
allowing researchers to design, experiment and visualise their own space
occupancy measure tailored for their specific project and helping them
understanding the ``null'' behaviour of the measures of interest.

\subsection{Space occupancy measures}\label{space-occupancy-measures}

\textcolor{blue}{Here we define trait spaces as any matrix where rows are observations and columns are their related traits.}
These traits can vary in number and types: they could be discrete
(e.g.~presence or absence of a bone; Beck and Lee 2014; Wright 2017),
continuous measurements (e.g.~leaf area; Díaz et al. 2016) or more
sophisticated ones (e.g.~landmark position; Marcy et al. 2016). Traits
can also be measured by using relative observations (e.g.~community
compositions; Jones et al. 2015) or distance between observations (e.g.
Close et al. 2015). However, regardless of the methodology used to build
a \textcolor{blue}{trait space}, three broad occupancy
measures can be used: the \textcolor{blue}{size} which
approximates the amount of space occupied, the density which
approximates the distribution in space and the position which
approximates the location in space (Fig. 1; Villéger et al. 2008). Of
course any combination of these three aspects is always possible.

\renewcommand\baselinestretch{1}\selectfont

\begin{figure}
\centering
\includegraphics{shiftingspace_files/figure-latex/fig_measures_types-1.pdf}
\caption{different type of information captured by space
occupancy measures: (A) size, (B) density and (C() position.}
\end{figure}

\renewcommand\baselinestretch{1.6}\selectfont

\paragraph{1. Size}\label{size}

\textcolor{blue}{Size captures the spread of a group in the trait space}.
They can be interpreted as the amount of the
\textcolor{blue}{trait space} that is occupied by
observations. Typically, larger values for such measures indicate the
presence of more extreme trait combinations. For example, if group A is
bigger than B, the observations in A achieve more extreme trait
combinations than in B. This type of measure is widely used in both
ecology (e.g.~the hypervolume; Blonder 2018) and in evolution (e.g.~the
sum or product of ranges or variances; Wills 2001).

Although
\textcolor{blue}{size measures are suitable indicators of a group's trait space}
occupancy, they are limited to comparing the range of trait-combinations
between groups.
\textcolor{blue}{Size measures do not take into account the distribution of the observations within a group and can often be insensitive to unoccupied "holes" in the trait space (overstimating the size; @blonder2018).}
They can make it difficult to determine whether all the observations are
on the edge of the group's distribution or whether the
\textcolor{blue}{size} is simply driven by outliers.

\paragraph{2. Density}\label{density}

\textcolor{blue}{Density gives an indication of the quantity of observations in the trait space.
They can be interpreted as the distribution of the observations *within* a group in the trait space.
Groups with higher density contain more observations that will tend to be more similar to each other.
For example, if group A is greater is size than group B and both have the same density, similar mechanisms could be driving both groups' trait space occupancy. 
This pattern could suggest that A is older and has had more time to achieve extreme trait combinations under essentially the same process as younger, smaller group B [@endler2005].
Note that density based measures can be sensitive to sampling (e.g. if only living species are present).
Density measures are less common compared to size measures, but they are still used in both ecology [e.g. the minimum spanning tree length; @oksanen2007vegan] and evolution [e.g. the average pairwise distance; @geiger2008]. }

\paragraph{3. Position}\label{position}

Position captures where a group lies in
\textcolor{blue}{trait space}. They can be interpreted as
where a group lies in the \textcolor{blue}{trait space}
either relative to the space itself or relative to another group. For
example, if group A has a different position than group B, A will have a
different trait-combination than in B.

Position measures may be harder to interpret in multidimensional spaces.
\textcolor{blue}{In a 2D space, two groups can be equally distant from a fixed point but in different parts of the space (left, right, up, or down - with the amount of parts of space increasing with dimensions).}
However, when thinking about unidimensional data, this measure is
obvious: two groups A or B could have the same variance
(\textcolor{blue}{size}) with the same number of
observations (density) but could have two different means and thus be in
different positions. These measures are used in ecology to compare the
position of two groups relative to each other (Mammola 2019).

\subsection{No measure to rule them all: benefits of considering
multiple
measures}\label{no-measure-to-rule-them-all-benefits-of-considering-multiple-measures}

The use of multiple measurements to assess
\textcolor{blue}{trait space} occupancy has the benefit of
providing a more detailed characterisation of occupancy changes. If the
question is to look at how space occupancy changes in response to mass
extinction, using a single space occupancy measure can miss part of the
picture: a change in
\textcolor{blue}{size could be decoupled from a change in position or density in trait space}.
For example, the
\textcolor{blue}{Cretaceous-Paleogene extinction (66 million years ago) shows an increase in size of the mammalian trait space}
(adaptive radiation; Halliday and Goswami 2016) but more specific
questions can be answered by looking at other aspects of
\textcolor{blue}{trait space occupancy: does the radiation expand}
on previously existing morphologies (elaboration, increase in density;
Endler et al. 2005) or does it explore new regions of the
\textcolor{blue}{trait space} (innovation, change in
position; Endler et al. 2005)? Similarly, in ecology, if two groups have
the same \textcolor{blue}{trait space size}, it can be
interesting to look at differences in density within these two groups:
different selection pressure can lead to different density within
equally \textcolor{blue}{sized} groups.

Here, we provide the first interdisciplinary review of 25 space
occupancy measures that uses the broad classification into
\textcolor{blue}{size, density and position to capture pattern changes in trait space}.
We assess the behaviour of measures using simulations and six
interdisciplinary empirical datasets covering a range of potential data
types and biological questions. We also introduce a tool for measuring
occupancy in multidimensional space
(\href{https://tguillerme.shinyapps.io/moms/}{\texttt{moms}}), which is
tailored to test the behaviour of measures for any use case.
\href{https://tguillerme.shinyapps.io/moms/}{\texttt{moms}} allows
workers to comprehensively assess the properties of their
\textcolor{blue}{trait space} and the measures associated
with their specific biological question.

\section{Methods}\label{methods}

We tested how 25 space occupancy measures relate to each other, are
affected by modifications of traits space and affect group comparisons
in empirical data:

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  We simulated 13 different spaces with different sets of parameters;
\item
  We transformed these spaces by removing 50\% of the observations
  following four different scenarios corresponding to different
  empirical scenarios: randomly, by limit (e.g.~expansion or reduction
  of niches), by density (e.g.~different degrees of competition within a
  guild) and by position (e.g.~ecological niche shift).
\item
  We measured occupancy on the resulting transformed spaces using eight
  different space occupancy measures;
\item
  We applied the same space occupancy measures to six empirical datasets
  (covering a range of disciplines and a range of dataset properties).
\end{enumerate}

\textcolor{blue}{Note that the paper contains the results for only eight measures which were selected as representative of common measures covering the size, density and position trait space aspects.}
The results for an additional 17 measures is available in the
supplementary material 4.

\subsection{Generating spaces}\label{generating-spaces}

We generated \textcolor{blue}{trait spaces} using the
following combinations of size, distributions, variance and correlation:

\renewcommand\baselinestretch{1}\selectfont

\begin{longtable}[]{@{}lllll@{}}
\toprule
\begin{minipage}[b]{0.12\columnwidth}\raggedright\strut
space name\strut
\end{minipage} & \begin{minipage}[b]{0.12333\columnwidth}\raggedright\strut
size\strut
\end{minipage} & \begin{minipage}[b]{0.31\columnwidth}\raggedright\strut
distribution(s)\strut
\end{minipage} & \begin{minipage}[b]{0.21\columnwidth}\raggedright\strut
dimensions variance\strut
\end{minipage} & \begin{minipage}[b]{0.13\columnwidth}\raggedright\strut
correlation\strut
\end{minipage}\tabularnewline
\midrule
\endhead
\begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
3D uniform\strut
\end{minipage} & \begin{minipage}[t]{0.12333\columnwidth}\raggedright\strut
200*3\strut
\end{minipage} & \begin{minipage}[t]{0.31\columnwidth}\raggedright\strut
Uniform (min = -0.5, max = 0.5)\strut
\end{minipage} & \begin{minipage}[t]{0.21\columnwidth}\raggedright\strut
Equal\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
None\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
15D uniform\strut
\end{minipage} & \begin{minipage}[t]{0.12333\columnwidth}\raggedright\strut
200*15\strut
\end{minipage} & \begin{minipage}[t]{0.31\columnwidth}\raggedright\strut
Uniform\strut
\end{minipage} & \begin{minipage}[t]{0.21\columnwidth}\raggedright\strut
Equal\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
None\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
50D uniform\strut
\end{minipage} & \begin{minipage}[t]{0.12333\columnwidth}\raggedright\strut
200*50\strut
\end{minipage} & \begin{minipage}[t]{0.31\columnwidth}\raggedright\strut
Uniform\strut
\end{minipage} & \begin{minipage}[t]{0.21\columnwidth}\raggedright\strut
Equal\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
None\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
150D uniform\strut
\end{minipage} & \begin{minipage}[t]{0.12333\columnwidth}\raggedright\strut
200*150\strut
\end{minipage} & \begin{minipage}[t]{0.31\columnwidth}\raggedright\strut
Uniform\strut
\end{minipage} & \begin{minipage}[t]{0.21\columnwidth}\raggedright\strut
Equal\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
None\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
50D uniform correlated\strut
\end{minipage} & \begin{minipage}[t]{0.12333\columnwidth}\raggedright\strut
200*50\strut
\end{minipage} & \begin{minipage}[t]{0.31\columnwidth}\raggedright\strut
Uniform\strut
\end{minipage} & \begin{minipage}[t]{0.21\columnwidth}\raggedright\strut
Equal\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
Random (between 0.1 and 0.9)\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
3D normal\strut
\end{minipage} & \begin{minipage}[t]{0.12333\columnwidth}\raggedright\strut
200*3\strut
\end{minipage} & \begin{minipage}[t]{0.31\columnwidth}\raggedright\strut
Normal (mean = 0, sd = 1)\strut
\end{minipage} & \begin{minipage}[t]{0.21\columnwidth}\raggedright\strut
Equal\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
None\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
15D normal\strut
\end{minipage} & \begin{minipage}[t]{0.12333\columnwidth}\raggedright\strut
200*15\strut
\end{minipage} & \begin{minipage}[t]{0.31\columnwidth}\raggedright\strut
Normal\strut
\end{minipage} & \begin{minipage}[t]{0.21\columnwidth}\raggedright\strut
Equal\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
None\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
50D normal\strut
\end{minipage} & \begin{minipage}[t]{0.12333\columnwidth}\raggedright\strut
200*50\strut
\end{minipage} & \begin{minipage}[t]{0.31\columnwidth}\raggedright\strut
Normal\strut
\end{minipage} & \begin{minipage}[t]{0.21\columnwidth}\raggedright\strut
Equal\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
None\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
150D normal\strut
\end{minipage} & \begin{minipage}[t]{0.12333\columnwidth}\raggedright\strut
200*150\strut
\end{minipage} & \begin{minipage}[t]{0.31\columnwidth}\raggedright\strut
Normal\strut
\end{minipage} & \begin{minipage}[t]{0.21\columnwidth}\raggedright\strut
Equal\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
None\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
50D normal correlated\strut
\end{minipage} & \begin{minipage}[t]{0.12333\columnwidth}\raggedright\strut
200*50\strut
\end{minipage} & \begin{minipage}[t]{0.31\columnwidth}\raggedright\strut
Normal\strut
\end{minipage} & \begin{minipage}[t]{0.21\columnwidth}\raggedright\strut
Equal\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
Random (between 0.1 and 0.9)\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
50D with random distributions\strut
\end{minipage} & \begin{minipage}[t]{0.12333\columnwidth}\raggedright\strut
200*50\strut
\end{minipage} & \begin{minipage}[t]{0.31\columnwidth}\raggedright\strut
Normal, Uniform, Lognormal (meanlog = 0, sdlog = 1)\strut
\end{minipage} & \begin{minipage}[t]{0.21\columnwidth}\raggedright\strut
Equal\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
None\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
50D PCA-like\strut
\end{minipage} & \begin{minipage}[t]{0.12333\columnwidth}\raggedright\strut
200*50\strut
\end{minipage} & \begin{minipage}[t]{0.31\columnwidth}\raggedright\strut
Normal\strut
\end{minipage} & \begin{minipage}[t]{0.21\columnwidth}\raggedright\strut
Multiplicative\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
None\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
50D PCO-like\strut
\end{minipage} & \begin{minipage}[t]{0.12333\columnwidth}\raggedright\strut
200*50\strut
\end{minipage} & \begin{minipage}[t]{0.31\columnwidth}\raggedright\strut
Normal\strut
\end{minipage} & \begin{minipage}[t]{0.21\columnwidth}\raggedright\strut
Additive\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
None\strut
\end{minipage}\tabularnewline
\bottomrule
\caption{different simulated space distribution.
\textcolor{blue}{ \textit{Name} of the simulated space; \textit{dimensions} of the matrix (row * columns); \textit{distribution(s)} of the data on each dimensions (for the 'Random', dimensions are randomly chosen between Normal, Uniform or Lognormal); \textit{dimension variance}: distribution of the variance between dimensions (when equal, the dimensions have the same variance); \textit{correlation} between dimensions.}}
\end{longtable}

\renewcommand\baselinestretch{1.6}\selectfont

The differences in \textcolor{blue}{trait space} sizes (200
elemeents for 3, 15, 50 or 150 dimensions) reflects the range found in
literature (e.g.~hopkins2017; Mammola 2019). We used a range of
distributions (uniform, normal or
\textcolor{blue}{a random combination of uniform, normal and lognormal})
to test the effect of observation distributions on the measurements. We
used different levels of variance for each dimensions in the spaces by
making the variance on each dimension either equal
(\(\sigma_{D1} \simeq \sigma_{D2} \simeq \sigma_{Di}\)) or decreasing
(\(\sigma_{D1} < \sigma_{D2} < \sigma_{Di}\)) with the decreasing factor
being either multiplicative (using the cumulative product of the inverse
of the number of dimensions: \(\prod_i^d(1/d)\)) or additive
(\(\sum_i^d(1/d)\)). Both reductions of variance are used to illustrate
the properties of ordinations where the variance decreases per
dimensions (and normal win Multidimensional Scaling - MDS, PCO or PCoA;
e.g. Close et al. 2015; lognormal in principal components analysis -
PCA; e.g. Marcy et al. 2016; Wright 2017; Healy et al. 2019). Finally,
we added a correlation parameter
\textcolor{blue}{to illustrate the effect of co-linearity between traits (especially in non-ordinated trait spaces).}
We repeated the simulation of each
\textcolor{blue}{trait space} 20 times (resulting in 260
spaces).

\subsection{Spatial occupancy
measures}\label{spatial-occupancy-measures}

We then
\textcolor{blue}{ calculated eight different measures } on
the resulting transformed spaces, including a new one, the average
displacement, which we expect to be influenced by changes in
\textcolor{blue}{trait space} position.

\renewcommand\baselinestretch{1}\selectfont

\begin{longtable}[]{@{}lllll@{}}
\toprule
\begin{minipage}[b]{0.12333\columnwidth}\raggedright\strut
Name\strut
\end{minipage} & \begin{minipage}[b]{0.23333\columnwidth}\raggedright\strut
Definition\strut
\end{minipage} & \begin{minipage}[b]{0.08333\columnwidth}\raggedright\strut
Captures\strut
\end{minipage} & \begin{minipage}[b]{0.11\columnwidth}\raggedright\strut
Source\strut
\end{minipage} & \begin{minipage}[b]{0.26\columnwidth}\raggedright\strut
Notes\strut
\end{minipage}\tabularnewline
\midrule
\endhead
\begin{minipage}[t]{0.12333\columnwidth}\raggedright\strut
Average Euclidean distance from centroid\strut
\end{minipage} & \begin{minipage}[t]{0.23333\columnwidth}\raggedright\strut
\(\frac{\sqrt{\sum_{i}^{n}{({k}_{n}-Centroid_{k})^2}}}{d}\)\strut
\end{minipage} & \begin{minipage}[t]{0.08333\columnwidth}\raggedright\strut
Size\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
Laliberté and Legendre (2010)\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
the functional dispersion (FDis - without abundance)\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.12333\columnwidth}\raggedright\strut
Sum of variances\strut
\end{minipage} & \begin{minipage}[t]{0.23333\columnwidth}\raggedright\strut
\(\sum_{i}^{d}{\sigma^{2}{k_i}}\)\strut
\end{minipage} & \begin{minipage}[t]{0.08333\columnwidth}\raggedright\strut
Size\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
Foote (1992)\strut
\end{minipage} & \begin{minipage}[t]{0.26\columnwidth}\raggedright\strut
common measure used in palaeobiology (Ciampaglio et al. 2001; Wills
2001)\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.12333\columnwidth}\raggedright\strut
Sum of ranges\strut
\end{minipage} & \begin{minipage}[t]{0.23333\columnwidth}\raggedright\strut
\(\sum_{i}^{d}{\|\text{max}(d_{i})-\text{min}(d_{i})\|}\)\strut
\end{minipage} & \begin{minipage}[t]{0.08333\columnwidth}\raggedright\strut
Size\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
Foote (1992)\strut
\end{minipage} & \begin{minipage}[t]{0.26\columnwidth}\raggedright\strut
more sensitive to outliers than the sum of variances\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.12333\columnwidth}\raggedright\strut
Ellipsoid volume\strut
\end{minipage} & \begin{minipage}[t]{0.23333\columnwidth}\raggedright\strut
\(\frac{\pi^{d/2}}{\Gamma(\frac{d}{2}+1)}\displaystyle\prod_{i}^{d} (\lambda_{i}^{0.5})\)\strut
\end{minipage} & \begin{minipage}[t]{0.08333\columnwidth}\raggedright\strut
Size\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
Donohue et al. (2013)\strut
\end{minipage} & \begin{minipage}[t]{0.26\columnwidth}\raggedright\strut
less sensitive to outliers than the convex hull hypervolume (Díaz et al.
2016; Blonder 2018)\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.12333\columnwidth}\raggedright\strut
Minimum spanning tree average distance\strut
\end{minipage} & \begin{minipage}[t]{0.23333\columnwidth}\raggedright\strut
\(\frac{\sum(\text{branch length})}{n}\)\strut
\end{minipage} & \begin{minipage}[t]{0.08333\columnwidth}\raggedright\strut
Density\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
Sedgewick (1990)\strut
\end{minipage} & \begin{minipage}[t]{0.26\columnwidth}\raggedright\strut
similar to the unscaled functional evenness (Villéger et al. 2008)\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.12333\columnwidth}\raggedright\strut
Minimum spanning tree distances evenness\strut
\end{minipage} & \begin{minipage}[t]{0.23333\columnwidth}\raggedright\strut
\(\frac{\sum\text{min}\left(\frac{\text{branch length}}{\sum\text{branch length}}\right)-\frac{1}{n-1}}{1-\frac{1}{n-1}}\)\strut
\end{minipage} & \begin{minipage}[t]{0.08333\columnwidth}\raggedright\strut
Density\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
Villéger et al. (2008)\strut
\end{minipage} & \begin{minipage}[t]{0.26\columnwidth}\raggedright\strut
the functional evenness without weighted abundance (FEve; Villéger et
al. 2008)\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.12333\columnwidth}\raggedright\strut
Average nearest neighbour distance\strut
\end{minipage} & \begin{minipage}[t]{0.23333\columnwidth}\raggedright\strut
\(\sqrt{\sum_{i}^{n}{min({q}_{i}-p_{i})^2}})\times \frac{1}{n}\)\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
Density\strut
\end{minipage} & \begin{minipage}[t]{0.08333\columnwidth}\raggedright\strut
Foote (1992)\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
the density of pairs of observations\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.12333\columnwidth}\raggedright\strut
Average displacement\strut
\end{minipage} & \begin{minipage}[t]{0.23333\columnwidth}\raggedright\strut
\(\frac{\sqrt{\sum_{i}^{n}{({k}_{n})^2}}}{\sqrt{\sum_{i}^{n}{({k}_{n}-Centroid_{k})^2}}}\)\strut
\end{minipage} & \begin{minipage}[t]{0.08333\columnwidth}\raggedright\strut
Position\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
This paper\strut
\end{minipage} & \begin{minipage}[t]{0.26\columnwidth}\raggedright\strut
the ratio between the observations' position from their centroid and the
centre of the trait space (coordinates: 0, 0, 0, \ldots{}). A value of 1
indicates that the observations' centroid is the centre of the trait
space\strut
\end{minipage}\tabularnewline
\bottomrule
\caption{List of measures with \emph{n} being the number of
observations, \emph{d} the total number of dimensions, \emph{k} any
specific row in the matrix, \emph{Centroid} being their mean and
\(\sigma^{2}\) their variance. \(\Gamma\) is the Gamma distribution and
\(\lambda_{i}\) the \textcolor{blue}{eigenvalue} of each
dimension and \({q}_{i}\) and \(p_{i}\) are any pairs of coordinates.}
\end{longtable}

\renewcommand\baselinestretch{1.6}\selectfont

We selected these eight space occupancy measures to illustrate how they
capture different aspects of space occupancy (not as an expression of
our preference).
\textcolor{blue}{These measures are specific to Euclidean and isotropic trait spaces (which is not necessary for all measures).}
The supplementary material 4 contains the same analysis as described
below, performed on 17 measures. Furthermore,
\href{https://tguillerme.shinyapps.io/moms/}{\texttt{moms}} allows
exploration into the effect of many more measures as well as the
customisation of measures by combining them or using user-designed
functions.

\subsection{Measure comparisons}\label{measure-comparisons}

We compared the space occupancy measures correlations across all
simulations between each pair of measures to assess
\textcolor{blue}{their} captured signal (Villéger et al.
2008; Laliberté and Legendre 2010). We used the measures on the full 13
\textcolor{blue}{trait spaces} described above. We then
scaled the results and measured the pairwise Pearson correlation to test
whether measures were capturing a similar signals or not using the
\texttt{psych} package (Revelle 2018).

\subsection{Changing space}\label{changing-spaces}

To assess how the measures responded to changes within
\textcolor{blue}{trait spaces}, we removed 50\% of
observations each time using the following algorithms:

\begin{itemize}
\item
  \textbf{Randomly:} by randomly removing 50\% of observations (Fig.
  2-A). This reflects a ``null'' biological model of changes in
  \textcolor{blue}{trait space}: the case when observations
  are removed regardless of their intrinsic characteristics. For
  example, if diversity is reduced by 50\% but the
  \textcolor{blue}{space size} remains the same, there is a
  decoupling between diversity and space occupancy (Ruta et al. 2013).
  Our selected measures are expected to not be affected by this change.
\item
  \textbf{Limit:} by removing observations within a distance from the
  centre of the \textcolor{blue}{trait space} lower or
  greater than a radius \(\rho\) (where \(\rho\) is chosen such that
  50\% observations are selected) generating two limit removals:
  \emph{maximum} and \emph{minimum} (respectively in orange and blue;
  Fig. 2-B). This can reflect a strict selection model where
  observations with trait values below or above a threshold are removed
  leading to an expansion or a contraction of the
  \textcolor{blue}{trait space}.
  \textcolor{blue}{Size} measures are expected to be most
  affected by this change.
\item
  \textbf{Density:} by removing any pairs of point with a distance \(D\)
  from each other where (where \(D\) is chosen such that 50\%
  observations are selected) generating two density removals:
  \emph{high} and \emph{low} (respectively in orange and blue; Fig.
  2-C). This can reflect changes within groups in the
  \textcolor{blue}{trait space} due to ecological factors
  (e.g.~niche repulsion resulting in lower density; Grant and Grant
  2006). Density measures are expected to be most affected by this
  change.
\item
  \textbf{Position:} by removing points similarly as for \textbf{Limit}
  but using the distance from the furthest point from the centre
  generating two position removals: \emph{positive} and \emph{negative}
  (respectively in orange and blue; Fig. 2-D). This can reflect global
  changes in \textcolor{blue}{trait space} (e.g.~if an
  entire group remaining diverse but occupying a different niche).
  Position measures are expected to be most affected by this change.
\end{itemize}

The algorithm to select \(\rho\) or \(D\) is described in the
Supplementary material 1.

\renewcommand\baselinestretch{1}\selectfont

\begin{figure}
\centering
\includegraphics[width=0.7\textwidth]{shiftingspace_files/figure-latex/fig_reduce_space-1.pdf}
\caption{\small{this figure illustrates the different type of space
reduction and how this could affect the measures for the simulation
results displayed in table 5. Each panel displays two groups of 50\% of
the points each. Each group (orange and blue) are generated using the
following algorithm: A - randomly; B - by limit (maximum and minimum
limit); C - by density (high and low); and D - by position (positive and
negative). Panel E et F represents two typical display of the reduction
results displayed in Table 5: the dots represent the median space
occupancy values across all simulations for each scenario of trait space
change (Table 2), the solid and dashed line respectively the 50\% and
95\% confidence intervals. Results in grey are the random 50\% reduction
(panel A). Results in blue and orange represent the opposite scenarios
from panels B, C, and D. The displayed value is the amount of overlap
(Bhattacharrya Coefficient) between the blue or orange distributions and
the grey one. Panel E and F shows respectively the ``ideal'' and
``worst'' results for any type of measures, where the space occupancy
measurement respectively manages or fails to captures a specific type of
reduction (size, position or density; Table 5).}}
\end{figure}

\renewcommand\baselinestretch{1.6}\selectfont


\textcolor{blue}{Because occupancy measures are dependent on the space, we scaled and centred them between -1 and 1 to make them comparable (by subtracting the observed occupancy without reduction to all the measures of the reduced spaces and then divided it by the maximum observed occupancy).}
A value of 0 indicates no effect of the space reduction and \(>0\) and
\(<0\) respectively indicates an increase or decrease in the measure
value. We then measured the \textcolor{blue}{amount} of
overlap between the non-random removals (limit, density and position)
and the random removals using the Bhattacharrya Coefficient
(Bhattacharyya 1943).

\subsubsection{Measuring the effect of space and
dimensionality}\label{measuring-the-effect-of-space-and-dimensionality}

Distribution differences and the number of dimensions can have an effect
on the measure results. For example, in a normally distributed space, an
\textcolor{blue}{increase in density can often lead to a decrease in size (though this is not necessarily true if the space is log-normal or uniform)}.
High dimensional spaces (\textgreater{}10) are subject to the ``curse of
multidimensionality'' (Bellman 1957): data becomes sparser with
increasing number of dimensions.
\textcolor{blue}{This can have two main consequences: 1) the probability of overlap between two groups decreases as a product of the number of dimensions; and 2) the amount of samples needed to "fill" the spaces increases exponentially [see this interactive illustration by Toph Tucker](https://observablehq.com/@tophtucker/theres-plenty-of-room-in-the-corners).}
The ``curse'' can make the interpretation of high dimensional data
counter-intuitive. For example if a group expands in multiple dimensions
(i.e.~increase in \textcolor{blue}{size}), the actual
hypervolume \textcolor{blue}{($\prod_{i}^{d} range_{Di}$)}
can decrease (Fig. 3 and Tables 6, 7).

We measured the effect of space distribution and dimensionality using an
ANOVA (\(occupancy \sim distribution\) and
\(occupancy \sim dimensions\)) by using all spaces with 50 dimensions
and the uniform and normal spaces with equal variance and no correlation
with 3, 15, 50, 100 and 150 dimensions (Table 2) for testing
respectively the effect of distribution and dimensions. The results of
the ANOVAs (\textcolor{blue}{F and *p*-values}) are reported
in Table 5 (full results in supplementary material 3).

\subsection{Empirical examples}\label{empirical-examples}

We analysed the effect of the different space occupancy measures on six
different empirical studies covering a range of fields that employ
\textcolor{blue}{trait space} analyses. For each of these
studies we generated \textcolor{blue}{trait spaces} from the
data published with the papers. We divided each
\textcolor{blue}{trait spaces} into two
biologically-relevant groups and tested whether the measures
differentiated the groups in different ways. Both the grouping and the
questions \textcolor{blue}{were} based on a simplified
version of the topics of these papers (with no intention to re-analyse
the data and questions). The procedures to generate the data and the
groups varies between studies and is detailed in the supplementary
materials 2.

\renewcommand\baselinestretch{1}\selectfont


\begin{longtable}[]{@{}llllllll@{}}
\toprule
\begin{minipage}[b]{0.08444\columnwidth}\raggedright\strut
study\strut
\end{minipage} & \begin{minipage}[b]{0.08444\columnwidth}\raggedright\strut
field\strut
\end{minipage} & \begin{minipage}[b]{0.08444\columnwidth}\raggedright\strut
taxonomic group\strut
\end{minipage} & \begin{minipage}[b]{0.13\columnwidth}\raggedright\strut
traits\strut
\end{minipage} & \begin{minipage}[b]{0.11\columnwidth}\raggedright\strut
trait space\strut
\end{minipage} & \begin{minipage}[b]{0.08444\columnwidth}\raggedright\strut
size\strut
\end{minipage} & \begin{minipage}[b]{0.07\columnwidth}\raggedright\strut
groups\strut
\end{minipage} & \begin{minipage}[b]{0.15\columnwidth}\raggedright\strut
question\strut
\end{minipage}\tabularnewline
\midrule
\endhead
\begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
Beck and Lee (2014)\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
Palaeontology\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
Mammalia\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
discrete morphological phylogenetic data\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
Ordination of a distance matrix (PCO)\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
106*105\strut
\end{minipage} & \begin{minipage}[t]{0.07\columnwidth}\raggedright\strut
52 crown vs.~54 stem\strut
\end{minipage} & \begin{minipage}[t]{0.15\columnwidth}\raggedright\strut
Are crown mammals more disparate than stem mammals?\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
Wright (2017)\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
Palaeontology\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
Crinoidea\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
discrete morphological phylogenetic data\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
Ordination of a distance matrix (PCO)\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
42*41\strut
\end{minipage} & \begin{minipage}[t]{0.07\columnwidth}\raggedright\strut
16 before vs.~23 after\strut
\end{minipage} & \begin{minipage}[t]{0.15\columnwidth}\raggedright\strut
Is there a difference in disparity before and after the Ordovician mass
extinction?\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
Marcy et al. (2016)\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
Evolution\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
Rodentia\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
skull 2D landmark coordinates\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
Ordination of a Procrustes Superimposition (PCA)\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
454*134\strut
\end{minipage} & \begin{minipage}[t]{0.07\columnwidth}\raggedright\strut
225 \emph{Megascapheus} vs.~229 \emph{Thomomys}\strut
\end{minipage} & \begin{minipage}[t]{0.15\columnwidth}\raggedright\strut
Are two genera of gopher morphologically distinct?\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
Hopkins and Pearson (2016)\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
Evolution\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
Trilobita\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
3D landmark coordinates\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
Ordination of a Procrustes Superimposition (PCA)\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
46*46\strut
\end{minipage} & \begin{minipage}[t]{0.07\columnwidth}\raggedright\strut
36 adults vs.~10 juveniles\strut
\end{minipage} & \begin{minipage}[t]{0.15\columnwidth}\raggedright\strut
Are juvenile trilobites a subset of adult ones?\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
Jones et al. (2015)\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
Ecology\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
Plantae\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
Communities species compositions\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
Ordination of a Jaccard distance matrix (PCO)\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
48*47\strut
\end{minipage} & \begin{minipage}[t]{0.07\columnwidth}\raggedright\strut
24 aspens vs.~24 grasslands\strut
\end{minipage} & \begin{minipage}[t]{0.15\columnwidth}\raggedright\strut
Is there a difference in species composition between aspens and
grasslands?\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
Healy et al. (2019)\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
Ecology\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
Animalia\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
Life history traits\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
Ordination of continuous traits (PCA)\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
285*6\strut
\end{minipage} & \begin{minipage}[t]{0.07\columnwidth}\raggedright\strut
83 ecthotherms vs.~202 endotherms\strut
\end{minipage} & \begin{minipage}[t]{0.15\columnwidth}\raggedright\strut
Do endotherms have more diversified life history strategies than
ectotherms?\strut
\end{minipage}\tabularnewline
\bottomrule
\caption{details of the six empirical trait spaces.}
\end{longtable}

\renewcommand\baselinestretch{1.6}\selectfont

For each empirical \textcolor{blue}{trait space} we
bootstrapped each group 500 times (Guillerme 2018) and applied the eight
space occupancy measure to each pairs of groups. We then compared the
means of each groups using the Bhattacharrya Coefficient (Bhattacharyya
1943).

\section{Results}\label{results}

\subsection{Measure comparisons}\label{measure-comparisons-1}

\renewcommand\baselinestretch{1}\selectfont


\begin{figure}
\centering
\includegraphics{shiftingspace_files/figure-latex/fig_measure_correlation-1.pdf}
\caption{pairwise correlation between the scaled measures.
Numbers on the upper right corner are the Pearson correlations. The red
line are linear regressions (with the confidence intervals in grey).
Av.: average; dist.: distance; min.: minimum; span.: spanning.}
\end{figure}

\renewcommand\baselinestretch{1.6}\selectfont


All the measures were either positively correlated (Pearson correlation
of 0.99 for the average Euclidean distance from centroid and sum of
variance or 0.97 for the average nearest neighbour distance and minimum
spanning tree average length; Fig. 3) or somewhat correlated (ranging
from 0.66 for the sum of variances and the ellipsoid volume to -0.09
between the average displacement and the average Euclidean distance from
centroid; Fig. 3). All measures but the ellipsoid volume were normally
(or nearly normally) distributed (Fig. 3).

\subsection{Space shifting}\label{space-shifting}

\renewcommand\baselinestretch{1}\selectfont

\begin{longtable}[]{@{}llllll@{}}
\toprule
\begin{minipage}[b]{0.10\columnwidth}\raggedright\strut
Measure\strut
\end{minipage} & \begin{minipage}[b]{0.13\columnwidth}\raggedright\strut
Size change\strut
\end{minipage} & \begin{minipage}[b]{0.08444\columnwidth}\raggedright\strut
Density change\strut
\end{minipage} & \begin{minipage}[b]{0.13\columnwidth}\raggedright\strut
Position change\strut
\end{minipage} & \begin{minipage}[b]{0.17\columnwidth}\raggedright\strut
Distribution effect\strut
\end{minipage} & \begin{minipage}[b]{0.23333\columnwidth}\raggedright\strut
Dimensions effect\strut
\end{minipage}\tabularnewline
\midrule
\endhead
\begin{minipage}[t]{0.10\columnwidth}\raggedright\strut
Average Euclidean distance from centroid\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-1.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-2.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-3.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright\strut
F = 0.924 ; p = 0.449\strut
\end{minipage} & \begin{minipage}[t]{0.23333\columnwidth}\raggedright\strut
F = 0.322 ; p = 0.958\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.10\columnwidth}\raggedright\strut
Sum of variances\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-4.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-5.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-6.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright\strut
F = 1.285 ; p = 0.274\strut
\end{minipage} & \begin{minipage}[t]{0.23333\columnwidth}\raggedright\strut
F = 0.478 ; p = 0.873\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.10\columnwidth}\raggedright\strut
Sum of ranges\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-7.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-8.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-9.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright\strut
F = 11.119 ; p = \textless{}1e-3***\strut
\end{minipage} & \begin{minipage}[t]{0.23333\columnwidth}\raggedright\strut
F = 32.307 ; p = \textless{}1e-3***\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.10\columnwidth}\raggedright\strut
Ellipsoid volume\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-10.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-11.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-12.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright\strut
F = 7.215 ; p = \textless{}1e-3***\strut
\end{minipage} & \begin{minipage}[t]{0.23333\columnwidth}\raggedright\strut
F = 13.486 ; p = \textless{}1e-3***\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.10\columnwidth}\raggedright\strut
Minimum spanning tree average distance\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-13.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-14.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-15.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright\strut
F = 1.162 ; p = 0.326\strut
\end{minipage} & \begin{minipage}[t]{0.23333\columnwidth}\raggedright\strut
F = 0.998 ; p = 0.435\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.10\columnwidth}\raggedright\strut
Minimum spanning tree distances evenness\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-16.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-17.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-18.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright\strut
F = 8.152 ; p = \textless{}1e-3***\strut
\end{minipage} & \begin{minipage}[t]{0.23333\columnwidth}\raggedright\strut
F = 29.358 ; p = \textless{}1e-3***\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.10\columnwidth}\raggedright\strut
Average nearest neighbour distance\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-19.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-20.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-21.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright\strut
F = 1.478 ; p = 0.207\strut
\end{minipage} & \begin{minipage}[t]{0.23333\columnwidth}\raggedright\strut
F = 0.773 ; p = 0.626\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.10\columnwidth}\raggedright\strut
Average displacements\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-22.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.08444\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-23.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-24.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright\strut
F = 10.742 ; p = \textless{}1e-3***\strut
\end{minipage} & \begin{minipage}[t]{0.23333\columnwidth}\raggedright\strut
F = 26.829 ; p = \textless{}1e-3***\strut
\end{minipage}\tabularnewline
\bottomrule
\caption{Results of the effect of space reduction, space dimension
distributions and dimensions number of the different space occupancy
measures. See Fig. 2 for interpretation of the figures distributions and values. F-values for distribution effect and dimensions effect represents respectively the effect of the ANOVAs  space occupancy ~ distributions and space occupancy ~ dimension  represent the ratio of sum squared difference within and between groups (the higher, the more the factor has an effect on the measure) and associated  \textit{p}-values (0 '\*\*\*' 0.001 '\*\*' 0.01 '\*' 0.05 '.' 0.1 '' 1). This figure illustrates how different measures can be influenced by different aspects of changes in the trait space.\textcolor{blue}{ E.g. the Average Euclidean distance from centroid (row 1) captures mainly changes in size (column 1), but also captures changes in density (column 2) but does not capture changes in position (column 3)}.}
\end{longtable}

\renewcommand\baselinestretch{1.6}\selectfont


As expected, some different measures capture different aspects of space
occupancy. However, it can be hard to predict the behaviour of each
measure when 50\% of the observations are removed. We observe a clear
decrease in median metric in less than a third of the space reductions
(10/36).

In terms of change in \textcolor{blue}{size}, only the
average Euclidean distance from centroid and the sum of variances seem
to capture a clear change in both directions. In terms of change in
density, only the minimum spanning tree average distance and the average
nearest neighbour distance seem to capture a clear change in both
directions. And in terms of change in position, only the average
displacement metric seems to capture a clear change in direction (albeit
not in both directions).

\subsection{Empirical examples}\label{empirical-example}

\renewcommand\baselinestretch{1}\selectfont

\begin{longtable}[]{@{}lllllll@{}}
\toprule
\begin{minipage}[b]{0.09\columnwidth}\raggedright\strut
Measure\strut
\end{minipage} & \begin{minipage}[b]{0.11\columnwidth}\raggedright\strut
Beck and Lee 2014\strut
\end{minipage} & \begin{minipage}[b]{0.12\columnwidth}\raggedright\strut
Wright 2017\strut
\end{minipage} & \begin{minipage}[b]{0.13\columnwidth}\raggedright\strut
Marcy et al. 2016\strut
\end{minipage} & \begin{minipage}[b]{0.11\columnwidth}\raggedright\strut
Hopkins and Pearson 2016\strut
\end{minipage} & \begin{minipage}[b]{0.13\columnwidth}\raggedright\strut
Jones et al. 2015\strut
\end{minipage} & \begin{minipage}[b]{0.11\columnwidth}\raggedright\strut
Healy et al. 2019\strut
\end{minipage}\tabularnewline
\midrule
\endhead
\begin{minipage}[t]{0.09\columnwidth}\raggedright\strut
\textcolor{blue}{Comparisons (orange *vs.* blue)}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\textcolor{blue}{crown *vs.* stem mammals morphologies}\strut
\end{minipage} & \begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
\textcolor{blue}{crinoids morphologies before *vs.* after the end-Ordovician extinction}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\textcolor{blue}{*Megascapheus* *vs.* *Thomomys* skull shapes}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\textcolor{blue}{adults *vs.* juveniles trilobites cephalon shapes}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\textcolor{blue}{aspens *vs.* grasslands communities compositions}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\textcolor{blue}{ecthotherms *vs.* endotherms life history traits}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright\strut
Average Euclidean distance from centroid\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-1.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-9.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-17.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-25.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-33.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-41.pdf}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright\strut
Sum of variances\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-2.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-10.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-18.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-26.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-34.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-42.pdf}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright\strut
Sum of ranges\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-3.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-11.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-19.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-27.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-35.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-43.pdf}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright\strut
Ellipsoid volume\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-4.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-12.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-20.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-28.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-36.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-44.pdf}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright\strut
Minimum spanning tree average distance\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-5.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-13.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-21.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-29.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-37.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-45.pdf}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright\strut
Minimum spanning tree distances evenness\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-6.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-14.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-22.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-30.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-38.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-46.pdf}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright\strut
Average nearest neighbour distance\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-7.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-15.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-23.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-31.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-39.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-47.pdf}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright\strut
Average displacements\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-8.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-16.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-24.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-32.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-40.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-48.pdf}\strut
\end{minipage}\tabularnewline
\bottomrule
\caption{Comparisons of pairs of groups in different empirical trait
spaces. NAs are used for cases where space occupancy could not be
measured due to the curse of multidimensionality. The displayed values
are the \textcolor{blue}{amount} of overlap between both
groups (Bhattacharrya Coefficient).}
\end{longtable}

\renewcommand\baselinestretch{1.6}\selectfont

Similarly as for the simulated results, the empirical ones indicate that
there \textcolor{blue}{can be} no perfect one-size-fit all
measurement. For all eight measures
(\textcolor{blue}{except} the ellipsoid volume) we see
either one group or the other having a bigger mean than the other and no
consistent case where a group has a bigger mean than the other for all
the measures. For example, in the Beck and Lee (2014)'s dataset, there
is a clear \textcolor{blue}{difference in size} using the
average Euclidean distance from centroid or the sum of variances
(overlaps of respectively 0.175 and 0.159) but no overlap when measuring
the \textcolor{blue}{size} using the sum of ranges (0.966).
However, for the Hopkins and Pearson (2016)'s dataset, this pattern is
reversed (no clear differences for the average Euclidean distance from
centroid or the sum of variances - 0.701 and 0.865 respectively - but a
clear difference for the sum of ranges (0). For each dataset, the
absolute differences between each groups is not consistent depending on
the measures. For example, in Hopkins and Pearson (2016)'s dataset, the
orange group's mean is clearly higher than the blue one when measuring
the sum of ranges (0) and the inverse is true when measuring the average
displacement (0).

\section{Discussion}\label{discussion}

Here we tested 25 measures of \textcolor{blue}{trait space}
occupancy on simulated and empirical datasets to assess how each measure
captures changes in \textcolor{blue}{trait space size},
density and position. Our results show that the correlation between
measures can vary both within and between measure categories (Fig. 3),
highlighting the importance of understanding the measure classification
for the interpretation of results. Our simulations show that different
measures capture different types of
\textcolor{blue}{trait space} change (Table 5), meaning that
the use of multiple measures is important for comprehensive
interpretation of \textcolor{blue}{trait space} occupancy.
We also show that the choice of measure impacts the interpretation of
group differences in empirical datasets (Table 6).

\paragraph{Measures comparisons}\label{measures-comparisons}

Measures within the same category of
\textcolor{blue}{trait space} occupancy
(\textcolor{blue}{size}, density or position) do not have
the same level of correlation with each other. For example, the average
Euclidean distance from centroid (\textcolor{blue}{size}) is
highly correlated to the sum of variances
(\textcolor{blue}{size} - correlation of 0.99) and somewhat
correlated with the minimum spanning tree average distance (density -
correlation of 0.66) but poorly with the ellipsoid volume
(\textcolor{blue}{size} - correlation of 0.17) and the
minimum spanning tree distances evenness (density - correlation of
-0.05).

\paragraph{Space shifting}\label{caveats}

Most measures capture no changes in space occupancy for the ``null''
(random) space reduction (in grey in Table 5). This is a desirable
behaviour for space occupancy measures since it will likely avoid false
positive errors in studies that estimate biological processes from space
occupancy patterns (e.g.~convergence Marcy et al. 2016, life history
traits Healy et al. (2019)). However, the average nearest neighbour
distance and the sum of ranges have a respectively positive and negative
``null'' median. This is not especially a bad property but it should be
kept in mind that even random processes can increase or decrease these
measures' values\}.

For changes in \textcolor{blue}{size}, the sum of variances
and the average Euclidean distance from centroid are good descriptors
(Table 5). However, as illustrated in the 2D examples in Fig. 2-B only
the blue change results (Table 5) should not result in a direct change
in \textcolor{blue}{overall size because the trait space} is
merely ``hollowed'' out. .

The average nearest neigbhour distance and the minimum spanning tree
average distance consistently detect changes in density with more
precision for low density \textcolor{blue}{trait spaces} (in
blue in Table 5). However, we can observe some degree of correlation
between the changes in density and the changes in
\textcolor{blue}{size} for most measure picking either
signal. This could be due to the use of normally distributed spaces
where a change in density often leads to a change in
\textcolor{blue}{size}. This is not necessarily the case
with empirical data.

Regarding the changes in position, only the average displacement measure
seems able to distinguish between a random change and a displacement of
the \textcolor{blue}{trait space} (Table 5).
\textcolor{blue}{However}, the average displacement measure
does not distinguish between positive or negative displacement: this
might be due to the inherent complexity of \emph{position} in a
multidimensional \textcolor{blue}{trait space}.

\paragraph{Empirical examples}\label{empirical-examples-1}

Although most differences are fairly consistent within each dataset with
one group having a higher space occupancy score than the other for
multiple measures, this difference can be more or less pronounced within
each dataset (ranging from no to nearly full overlap - BC
\(\in(0;0.995)\)) and sometimes even reversed. This indicates that
opposite conclusions can be drawn from a dataset depending on which
space occupancy measure is considered. These differences depending on
the measures are also more pronounced in the empirical datasets where
the observations per group are unequal (Hopkins and Pearson 2016; Healy
et al. 2019).

\subsubsection{Caveats}\label{caveats}

While our simulations are useful to illustrate the behaviour of diverse
space occupancy measures, they have several caveats. First, the
simulated observations in the \textcolor{blue}{trait spaces}
are independent. This is not the case in biology where observations can
be spatially (Jones et al. 2015) or phylogenetically correlated (e.g.
Beck and Lee 2014). Second, the algorithm used to reduce the
\textcolor{blue}{trait spacesmight not always accurately reflect changes}.
This might favour some specific measures over others, in particular for
the changes in density that modify the nearest neighbour density rather
than changing the global density. This algorithmic choice was made in
order to not confound changes in density along with changes in
\textcolor{blue}{size}. However, the results presented here
probably capture the general behaviour of each measure since results are
consistent between the simulated and empirical analysis. Furthermore,
\href{https://tguillerme.shinyapps.io/moms/}{\texttt{moms}} allows
workers to test the caveats mentioned above by uploading empirical
\textcolor{blue}{trait spaces}.

\subsubsection{Conclusions}\label{conclusions}

We insist that no measure is better than the next one and that workers
should identify the most appropriate measures based on their
\textcolor{blue}{trait space} properties as well as their
specific biological question. However, following the findings of this
study we make several suggestions:

First, we suggest using multiple measures to tackle different aspects of
the \textcolor{blue}{trait space}. Although using multiple
measures is not uncommon in macroevolutionary studies (e.g. Halliday and
Goswami 2016) or in ecology (Mammola 2019),
\textcolor{blue}{they often do no cover more than one of the three categories of trait space measures}.

Second, we suggest selecting the measures that best address the
biological question \textcolor{blue}{at} hand. If one
studies an adaptive radiation in a group of organisms, it is worth
thinking what would be the expected null model: would the group's
\textcolor{blue}{size} increase (radiation in all
directions), would it increase in density (niche specialisation) or
would it shift in position (radiation into a new set of niches)?

Third, we suggest not naming measures after the biological aspect they
describe
\textcolor{blue}{which can be vague (e.g. "disparity" or "functional dispersion") but rather after what they are measuring and why (e.g. "we used sum of ranges to measure the space size").}
We believe this will support both a clearer understanding of what
\emph{is} measured as well as better communication between ecology and
evolution research where measures can be similar but have different
names.

Multidimensional analyses have been acknowledged as essential tools in
modern biology but they can often be counter-intuitive (Bellman 1957).
It is thus crucial to accurately describe patterns in multidimensional
\textcolor{blue}{trait spaces} to be able to link them to
biological processes. When summarising
\textcolor{blue}{trait spaces}, it is important to remember
that a pattern captured by a specific space occupancy measure is often
dependent on the properties of the space and of the particular
biological question of interest. We believe that having a clearer
understanding of both the properties of the
\textcolor{blue}{trait space} and the associated space
occupancy measures (e.g.~using
\href{https://tguillerme.shinyapps.io/moms/}{\texttt{moms}}) as well as
using novel space occupancy measures to answer specific questions will
be of great use to study biological processes in a multidimensional
world.

\section{Acknowledgements}\label{acknowledgements}

We thank Natalie Jones and Kevin Healy for helping with the empirical
datasets and two anonymous reviewer for their comments. We acknowledge
funding from the Australian Research Council DP170103227 and FT180100634
awarded to VW.

\section{Authors contributions}\label{authors-contributions}

TG, MNP, AEM and VW designed the project. TG and AEM collected the
empirical dataset. TG ran the analyses and designed the software. TG,
MNP, AEM and VW wrote the manuscript.

\section{Data Availability, repeatability and
reproducibility}\label{data-availability-repeatability-and-reproducibility}

The raw empirical data is available from the original papers (Beck and
Lee 2014; Jones et al. 2015, Marcy et al. (2016); Hopkins and Pearson
2016; Wright 2017; Healy et al. 2019). The subsets of the empirical data
used in this analysis are available on figshare
\href{https://doi.org/10.6084/m9.figshare.9943181.v1}{DOI:
10.6084/m9.figshare.9943181.v1}. The modified empirical data are
available in the package accompanying this manuscript
(\texttt{data(moms::demo\_data)}). This manuscript (including the
figures, tables and supplementary material) is repeatable and
reproducible by compiling the vignette of the
\href{https://github/TGuillerme/moms}{GitHub \texttt{moms\ R} package}.

\section*{References}\label{references}
\addcontentsline{toc}{section}{References}

\hypertarget{refs}{}
\hypertarget{ref-beck2014}{}
Beck R.M.D., Lee M.S.Y. 2014. Ancient dates or accelerated rates?
Morphological clocks and the antiquity of placental mammals. Proceedings
of the Royal Society B: Biological Sciences. 281:20141278.

\hypertarget{ref-cursedimensionality}{}
Bellman R.E. 1957. Dynamic programming. Princeton University Press.

\hypertarget{ref-bhattacharyya1943}{}
Bhattacharyya A. 1943. On a measure of divergence between two
statistical populations defined by their probability distributions.
Bulletin of the Calcutta Mathematical Society. 35:99--109.

\hypertarget{ref-blonder2018}{}
Blonder B. 2018. Hypervolume concepts in niche-and trait-based ecology.
Ecography. 41:1441--1455.

\hypertarget{ref-ciampaglio2001}{}
Ciampaglio C.N., Kemp M., McShea D.W. 2001. Detecting changes in
morphospace occupation patterns in the fossil record: Characterization
and analysis of measures of disparity. Paleobiology. 71:695--715.

\hypertarget{ref-close2015}{}
Close R.A., Friedman M., Lloyd G.T., Benson R.B. 2015. Evidence for a
mid-Jurassic adaptive radiation in mammals. Current Biology.

\hypertarget{ref-diaz2016}{}
Díaz S., Kattge J., Cornelissen J.H., Wright I.J., Lavorel S., Dray S.,
Reu B., Kleyer M., Wirth C., Prentice I.C., others. 2016. The global
spectrum of plant form and function. Nature. 529:167.

\hypertarget{ref-donohue2013}{}
Donohue I., Petchey O.L., Montoya J.M., Jackson A.L., McNally L., Viana
M., Healy K., Lurgi M., O'Connor N.E., Emmerson M.C. 2013. On the
dimensionality of ecological stability. Ecology Letters. 16:421--429.

\hypertarget{ref-endler2005}{}
Endler J.A., Westcott D.A., Madden J.R., Robson T. 2005. Animal visual
systems and the evolution of color patterns: Sensory processing
illuminates signal evolution. Evolution. 59:1795--1818.

\hypertarget{ref-foote1992}{}
Foote M. 1992. Rarefaction analysis of morphological and taxonomic
diversity. Paleobiology. 18:1--16.

\hypertarget{ref-grant2006}{}
Grant P.R., Grant B.R. 2006. Evolution of character displacement in
darwins finches. Science. 313:224--226.

\hypertarget{ref-disprity}{}
Guillerme T. 2018. dispRity: A modular R package for measuring
disparity. Methods in Ecology and Evolution. 9:1755--1763.

\hypertarget{ref-halliday2015}{}
Halliday T.J.D., Goswami A. 2016. Eutherian morphological disparity
across the end-cretaceous mass extinction. Biological Journal of the
Linnean Society. 118:152--168.

\hypertarget{ref-geiger2008}{}
Harmon L.J., Weir J.T., Brock C.D., Glor R.E., Challenger W. 2008.
GEIGER: Investigating evolutionary radiations. Bioinformatics.
24:129--131.

\hypertarget{ref-healy2019}{}
Healy K., Ezard T.H.G., Jones O.R., Salguero-G'omez R., Buckley Y.M.
2019. Animal life history is shaped by the pace of life and the
distribution of age-specific mortality and reproduction. Nature Ecology
\& Evolution. 2397-334X.

\hypertarget{ref-hopkins2016}{}
Hopkins M., Pearson K. 2016. Non-linear ontogenetic shape change in
cryptolithus tesselatus (trilobita) using three-dimensional geometric
morphometrics. Palaeontologia Electronica. 19:1--54.

\hypertarget{ref-hopkins2017}{}
Hopkins M.J., Gerber S. 2017. Morphological disparity. In: Nuno de la
Rosa L., Müller G., editors. Evolutionary developmental biology: A
reference guide. Cham: Springer International Publishing. p. 1--12.

\hypertarget{ref-jones2015}{}
Jones N.T., Germain R.M., Grainger T.N., Hall A.M., Baldwin L., Gilbert
B. 2015. Dispersal mode mediates the effect of patch size and patch
connectivity on metacommunity diversity. Journal of Ecology.
103:935--944.

\hypertarget{ref-lalibertuxe92010}{}
Laliberté É., Legendre P. 2010. A distance-based framework for measuring
functional diversity from multiple traits. Ecology. 91:299--305.

\hypertarget{ref-legendre2012}{}
Legendre P., Legendre L.F. 2012. Numerical ecology. Elsevier.

\hypertarget{ref-mammola2019}{}
Mammola S. 2019. Assessing similarity of n-dimensional hypervolumes:
Which metric to use? Journal of Biogeography. 0.

\hypertarget{ref-marcy2016}{}
Marcy A.E., Hadly E.A., Sherratt E., Garland K., Weisbecker V. 2016.
Getting a head in hard soils: Convergent skull evolution and divergent
allometric patterns explain shape variation in a highly diverse genus of
pocket gophers (thomomys). BMC evolutionary biology. 16:207.

\hypertarget{ref-oksanen2007vegan}{}
Oksanen J., Kindt R., Legendre P., O'Hara B., Stevens M.H.H., Oksanen
M.J., Suggests M. 2007. The vegan package. Community ecology package.
10:631--637.

\hypertarget{ref-psych}{}
Revelle W. 2018. Psych: Procedures for psychological, psychometric, and
personality research. Evanston, Illinois: Northwestern University.

\hypertarget{ref-ruta2013}{}
Ruta M., Angielczyk K.D., Fröbisch J., Benton M.J. 2013. Decoupling of
morphological disparity and taxic diversity during the adaptive
radiation of anomodont therapsids. Proceedings of the Royal Society of
London B: Biological Sciences. 280.

\hypertarget{ref-sedgewick1990}{}
Sedgewick R. 1990. Algorithms in c. Addison-Wesley, Reading.

\hypertarget{ref-villuxe9ger2008}{}
Villéger S., Mason N.W.H., Mouillot D. 2008. New multidimensional
functional diversity indices for a multifaceted framework in functional
ecology. Ecology. 89:2290--2301.

\hypertarget{ref-wills2001}{}
Wills M.A. 2001. Morphological disparity: A primer. In: Adrain J.M.,
Edgecombe G.D., Lieberman B.S., editors. Fossils, phylogeny, and form.
Springer US. p. 55--144.

\hypertarget{ref-wright2017}{}
Wright D.F. 2017. Phenotypic innovation and adaptive constraints in the
evolutionary radiation of palaeozoic crinoids. Scientific Reports.
7:13745.


\end{document}
