\documentclass[]{article}
\usepackage{lmodern}
\usepackage{lineno}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage[margin=1in]{geometry}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdftitle={Shifting spaces: which disparity or dissimilarity metrics best summarise occupancy in multidimensional spaces?},
            pdfauthor={Thomas Guillerme, Mark N. Puttick, Ariel E. Marcy, Vera Weisbecker},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{longtable,booktabs}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\providecommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}


\linespread{2}


\setlength{\droptitle}{-2em}

  \title{Shifting spaces: which disparity or dissimilarity metrics best summarise
occupancy in multidimensional spaces?}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
    \author{Thomas Guillerme$^{1,*}$, Mark N. Puttick$^{2}$, Ariel E. Marcy$^{1}$, Vera Weisbecker$^{1}$}
    \preauthor{\centering\large\emph}
  \postauthor{\par}
      \predate{\centering\large\emph}
  \postdate{\par}
    \date{2019-10-06}


\begin{document}

\modulolinenumbers[1] % just after the \begin{document} tag
\linenumbers

\maketitle

\noindent {\small \it 
$^1$School of Biological Sciences, University of Queensland, St. Lucia, Queensland, Australia.\\
$^2$School of Earth Sciences, University of Bristol, Wills Memorial Building, Queen's Road, Bristol BS8 1RJ, UK.\\}


\section{Abstract}\label{abstract}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Multidimensional analysis of traits are now a common toolkit in
  ecology and evolution and are based on trait-spaces in which each
  dimension summarise the observed trait combination (a morphospace or
  an ecospace). Observations of interest will typically occupy a subset
  of this trait-space, and researchers will apply one or more metrics to
  quantify the way in which organisms ``inhabit'' that trait-space. In
  macroevolution and ecology these metrics are referred to as disparity
  or dissimilarity metrics and can be generalised as space occupancy
  metrics. Researchers use these metrics to investigate how space
  occupancy changes through time, in relation to other groups of
  organisms, and in response to global environmental changes, such as
  global warming events or mass extinctions. However, the mathematical
  and biological meaning of most space occupancy metrics is vague with
  the majority of widely-used metrics lacking formal description.
\item
  Here we propose a broad classification of space occupancy metrics into
  three categories that capture changes in volume, density, or position.
  We analyse the behaviour of 25 metrics to study changes in trait-space
  volume, density and position on a series of simulated and empirical
  datasets.
\item
  We find no one metric describes all of trait-space but that some
  metrics are better at capturing certain aspects compared to other
  approaches and that their performance depends on both the trait-space
  and the hypothesis analysed. However, our results confirm the three
  broad categories (volume, density and position) and allow to relate
  changes in any of these categories to biological phenomena.
\item
  Since the choice of space occupancy metric should be specific to the
  data and question at had, we introduced
  \href{https://tguillerme.shinyapps.io/moms/}{\texttt{moms}}, a
  user-friendly tool based on a graphical interface that allows users to
  both visualise and measure changes space occupancy for any metric in
  simulated or imported trait-spaces. Users are also provided with tools
  to transform their data in space (e.g.~contraction, displacement,
  etc.). This tool is designed to help researchers choose the right
  space occupancy metrics, given the properties of their trait-space and
  their biological question.
\end{enumerate}

\section{Introduction}\label{introduction}

Groups of species and environments share specific, easily recognisable,
correlated characteristics of many kinds: guilds or biomes with shared
phenotypic, physiological, phylogenetic or behavioural traits. Organisms
or environments should therefore be studied as a set of traits rather
than some specific traits in isolation (Donohue et al. 2013; Hopkins and
Gerber 2017). Biologists have increasingly been using ordination
techniques (see Legendre and Legendre 2012 for a summary) to create
multidimensional trait-spaces to either explore properties of the data
or test hypotheses (Oksanen et al. 2007; Adams and Ot√°rola-Castillo
2013; Bonhomme et al. 2014; Blonder 2018; Guillerme 2018). For example,
in palaeobiology, Wright (2017) use trait-spaces to study how groups of
species' characteristics change through time; in ecology, Jones et al.
(2015) study evidence of competition by looking at trait overlap between
two populations. However, different fields use a different set of terms
for such approaches (Table 1). Nonetheless, they are the same
mathematical objects: matrices with columns representing an original or
transformed trait value and rows representing observations, such as
taxon, field site, etc. (Guillerme 2018).


\renewcommand\baselinestretch{1}\selectfont


\begin{longtable}[]{@{}llll@{}}
\toprule
\begin{minipage}[b]{0.20\columnwidth}\raggedright\strut
In mathematics\strut
\end{minipage} & \begin{minipage}[b]{0.16\columnwidth}\raggedright\strut
In ecology\strut
\end{minipage} & \begin{minipage}[b]{0.25\columnwidth}\raggedright\strut
In macroevolution\strut
\end{minipage} & \begin{minipage}[b]{0.20\columnwidth}\raggedright\strut
In this paper\strut
\end{minipage}\tabularnewline
\midrule
\endhead
\begin{minipage}[t]{0.20\columnwidth}\raggedright\strut
Matrix (\(n \times d\))\strut
\end{minipage} & \begin{minipage}[t]{0.16\columnwidth}\raggedright\strut
Function-space, Eco-space, etc.\strut
\end{minipage} & \begin{minipage}[t]{0.25\columnwidth}\raggedright\strut
Morphospace, traitspace, etc.\strut
\end{minipage} & \begin{minipage}[t]{0.20\columnwidth}\raggedright\strut
trait-space\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.20\columnwidth}\raggedright\strut
Rows (\emph{n})\strut
\end{minipage} & \begin{minipage}[t]{0.16\columnwidth}\raggedright\strut
Taxa, field sites, environments, etc.\strut
\end{minipage} & \begin{minipage}[t]{0.25\columnwidth}\raggedright\strut
Taxa, specimen, populations, etc.\strut
\end{minipage} & \begin{minipage}[t]{0.20\columnwidth}\raggedright\strut
observations\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.20\columnwidth}\raggedright\strut
Columns (\emph{d})\strut
\end{minipage} & \begin{minipage}[t]{0.16\columnwidth}\raggedright\strut
Traits, Ordination scores, distances, etc.\strut
\end{minipage} & \begin{minipage}[t]{0.25\columnwidth}\raggedright\strut
Traits, Ordination scores, distances, etc.\strut
\end{minipage} & \begin{minipage}[t]{0.20\columnwidth}\raggedright\strut
dimensions\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.20\columnwidth}\raggedright\strut
Matrix subset (\(m \times d\); \(m \leq n\))\strut
\end{minipage} & \begin{minipage}[t]{0.16\columnwidth}\raggedright\strut
Treatments, phylogenetic group (clade), etc.\strut
\end{minipage} & \begin{minipage}[t]{0.25\columnwidth}\raggedright\strut
Clades, geological stratum, etc.\strut
\end{minipage} & \begin{minipage}[t]{0.20\columnwidth}\raggedright\strut
group\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.20\columnwidth}\raggedright\strut
Statistic\strut
\end{minipage} & \begin{minipage}[t]{0.16\columnwidth}\raggedright\strut
Dissimilarity index or metric, hypervolume, functional diversity\strut
\end{minipage} & \begin{minipage}[t]{0.25\columnwidth}\raggedright\strut
Disparity metric or index\strut
\end{minipage} & \begin{minipage}[t]{0.20\columnwidth}\raggedright\strut
space occupancy metric\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.20\columnwidth}\raggedright\strut
Multidimensional analysis\strut
\end{minipage} & \begin{minipage}[t]{0.16\columnwidth}\raggedright\strut
Dissimilarity analysis, trait analysis, etc.\strut
\end{minipage} & \begin{minipage}[t]{0.25\columnwidth}\raggedright\strut
Disparity analysis, disparity-through-time, etc.\strut
\end{minipage} & \begin{minipage}[t]{0.20\columnwidth}\raggedright\strut
multidimensional analysis\strut
\end{minipage}\tabularnewline
\bottomrule
\end{longtable}

Table 1: terms and equivalence between mathematics, ecology and
macroevolution.

\renewcommand\baselinestretch{2}\selectfont

Ecologists and evolutionary biologists also often use trait-spaces with
respect to the same fundamental questions: are groups overlapping in the
trait-space? Are some regions of the trait-space not occupied? How do
specific factors influence the occupancy of the trait-space? Studying
the occupancy of trait-spaces can be achieved using disparity metrics in
macroevolution (Wills 2001; Hopkins and Gerber 2017; Guillerme 2018) or
comparing hypervolumes in ecology (Donohue et al. 2013; D√≠az et al.
2016; Blonder 2018; Mammola 2019). However, although these space
occupancy metrics are common in ecology and evolution, surprisingly
little work has been published on their behaviour (but see Ciampaglio et
al. 2001; Vill√©ger et al. 2008; Mammola 2019).

Different space occupancy metrics capture different aspects of the
trait-space (ciampaglio2001; Vill√©ger et al. 2008; Mammola 2019). It may
be widely-known by many in the field, but to our knowledge this is
infrequently mentioned in peer-reviewed papers. First, space occupancy
metrics are often named as the biological aspect they are describing
(e.g. ``disparity'' or ``functional diversity'') rather than what they
are measuring (e.g.~the average pairwise distances) which obscures the
differences or similarities between studies. Second, in many studies in
ecology and evolution, authors have focused on measuring the volume of
the trait-space with different metrics (e.g.~ellipsoid volume Donohue et
al. 2013; hypervolume D√≠az et al. 2016; Procrustes variance Marcy et al.
2016; product of variance Wright 2017). However, volume only represents
a single aspects of space occupancy, disregarding others such as the
density (Harmon et al. 2008) or position (Wills 2001; Ciampaglio et al.
2001). For example, if two groups occupy the same volume in trait-space,
this will lead to supporting certain biological conclusions. Yet, an
alternative aspect of space occupancy may indicate that the groups'
position are different; this would likely lead to a different biological
conclusion (e.g.~the groups are equally diverse but occupy different
niches). Using metrics that only measure one aspect of the
multidimensional trait-space may restrain the potential of
multidimensional analysis (Vill√©ger et al. 2008).

Here we propose a broad classification of space occupancy metrics as
used across ecology and evolution and analyse their power to detect
changes in trait-space occupancy in simulated and empirical data. We
provide an assessment of each broad type of space occupancy metrics
along with a unified terminology to foster communication between ecology
and evolution. Unsurprisingly, we found no one metric describes all
changes through a trait-space and the results from each metric are
dependent on the characteristics of the space and the hypotheses.
Furthermore, because there can potentially be an infinite number of
metrics, it would be impossible to propose clear generalities to space
occupancy metrics behavior. Therefore, we propose
\href{https://tguillerme.shinyapps.io/moms/}{\texttt{moms}}, a
user-friendly tool allowing researchers to design, experiment and
visualise their own space occupancy metric tailored for their specific
project and helping them understanding the ``null'' behavior of their
metrics of interest.

\subsection{Space occupancy metrics}\label{space-occupancy-metrics}

In this paper, we define trait-spaces as any matrix where rows are
observations and columns are traits. These traits can widely vary in
number and types: they could be coded as discrete (e.g.~presence or
absence of a bone; Beck and Lee 2014; Wright 2017), continuous
measurements (e.g.~leaf area; D√≠az et al. 2016) or more sophisticated
measures (Fourier ellipses; Bonhomme et al. 2014; e.g.~landmark
position; Marcy et al. 2016). Traits can also be measured by using
relative observations (e.g.~community compositions; Jones et al. 2015)
or distance between observations (e.g. Close et al. 2015). However,
regardless of the methodology used to build a trait-space, three broad
occupancy metrics can be measured: the volume which will approximate the
amount of space occupied, the density which will approximate the
distribution in space and the position which will approximate the
location in space (Fig. 1; Vill√©ger et al. 2008). Of course any
combination of these three aspects is always possible.

\renewcommand\baselinestretch{1}\selectfont
\begin{figure}
\centering
\includegraphics{shiftingspace_files/figure-latex/fig_metrics_types-1.pdf}
\caption{different type of information captured by space
occupancy metrics. A - Volume (e.g.~sum of ranges); B - Density
(e.g.~average squared pairwise distances); C - Position (e.g.~median
distance from centroid).}
\end{figure}
\renewcommand\baselinestretch{2}\selectfont



\paragraph{1. Volume}\label{volume}

Volume metrics measure the spread of a group in the trait-space. They
can be interpreted as the amount of the trait-space that is occupied by
observations. Typically, larger values for such metrics indicate the
presence of more extreme trait combinations. For example, if group A has
a bigger volume than group B, the observations in A achieve more extreme
trait combinations than in B. This type of metric is widely used in both
ecology (e.g.~the hypervolume; Blonder 2018) and in evolution (e.g.~the
sum or product of ranges or variances; Wills 2001).

Although volume metrics are a suitable indicator for comparing a group's
trait-space occupancy, it is limited to comparing the range of
trait-combinations between groups. Volume metrics do not take into
account the distribution of the observations within a group. In other
words, they can make it difficult to determine whether all the
observations are on the edge of the volume or whether the volume is
simply driven by small number of extreme observations.

\paragraph{2. Density}\label{density}

Density metrics measure the distribution of a group in the trait-space.
They can be interpreted as the distribution of the observations
\emph{within} a group in the trait-space. Groups with higher density
have observations within it that tend to be more similar to each other.
For example, if group A has a greater volume than group B but both have
the same density, similar mechanisms could be driving both groups'
trait-space occupancy. However, this might suggest that A is older and
had more time to achieve more extreme trait combinations under
essentially the same process (Endler et al. 2005). Density is less
commonly measured compared to volume, but it is still used in both
ecology (e.g.~the minimum spanning tree length; Oksanen et al. 2007) and
evolution (e.g.~the average pairwise distance; Harmon et al. 2008).

\paragraph{3. Position}\label{position}

Position metrics measure where a group lies in trait-space. They can be
interpreted as where a group lies in the trait-space either relative to
the space itself or relative to another group. For example, if group A
has a different position than group B, observations in A will have a
different trait-combination than B.

Position metrics may be harder to interpret in multidimensional spaces
(i.e.~beyond left/right, up/down and front/back). However, when thinking
about unidimensional data (one trait), this metric is obvious: two
groups A or B could have the same variance (i.e. ``volume'' or spread)
with the same number of observations (i.e.~density) but could have a
different mean and thus be in different positions. These metrics have
been used in ecology to compare the position of two groups relatively to
each other (Mammola 2019).

\subsection{No metric to rule them all: benefits of considering multiple
metrics}\label{no-metric-to-rule-them-all-benefits-of-considering-multiple-metrics}

The use of multiple metrics to assess trait-space occupancy has the
benefit of providing a more detailed characterisation of occupancy
changes. If the question of interest is, say, to look at how space
occupancy changes in response to mass extinction, using a single space
occupancy metric can miss part of the picture: a change in volume could
be decoupled from a change in position or density in trait-space. For
example, the Cretaceous-Palaeogene extinction (66 million years ago -
Mya) has been linked to an increase in volume of the mammalian
trait-space (adaptive radiation; Halliday and Goswami 2016) but more
specific questions can be answered by looking at other aspects of
trait-space occupancy: does the radiation expands on previously existing
morphologies (elaboration, increase in density; Endler et al. 2005) or
does it explore new regions of the trait-space (innovation, change in
position; Endler et al. 2005)? Similarly, in ecology, if two groups
occupy the same volume in the trait-space, it can be interesting to look
at differences in density within these two groups: different selection
pressure can lead to different density within equal volume groups.

Here, we provide the first interdisciplinary review of 25 space
occupancy metrics that uses the broad classification of metrics into
volume, density and position to capture pattern changes in trait-space.
We assess the behaviour of metrics using simulations and six
interdisciplinary empirical datasets covering a wide range of potential
data types and biological questions. We also introduce a tool for
measuring occupancy in multidimensional space
(\href{https://tguillerme.shinyapps.io/moms/}{\texttt{moms}}), which is
a user-friendly, open-source, graphical interface to allow the tailored
testing of metric behaviour for any use case.
\href{https://tguillerme.shinyapps.io/moms/}{\texttt{moms}} will allow
workers to comprehensively assess the properties of their trait-space
and the metrics associated with their specific biological question.

\section{Methods}\label{methods}

We tested how 25 different space occupancy metrics relate to each other,
are affected by modifications of traits space and affect group
comparisons in empirical data. To do so, we performed the following
steps (explained in more detail below):

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  We simulated 13 different spaces with different sets of parameters;
\item
  We transformed these spaces by removing 50\% of the observations
  following four different scenarios corresponding to different
  empirical scenarios: randomly, by limit (e.g.~expansion or reduction
  of niches), by density (e.g.~different degrees of competition within a
  guild) and by position (e.g.~ecological niche shift).
\item
  We measured occupancy on the resulting transformed spaces using eight
  different space occupancy metrics;
\item
  We applied the same space occupancy metrics to six empirical datasets
  (covering a range of disciplines and a range of dataset properties).
\end{enumerate}

Note that the paper contains the results for only eight metrics, the
results for the additional 17 metrics is available in the supplementary
material 4.

\subsection{Generating spaces}\label{generating-spaces}

We generated trait-spaces using the following combinations of size,
distributions, variance and correlation:

\renewcommand\baselinestretch{1}\selectfont

\begin{longtable}[]{@{}lllll@{}}
\toprule
space name & size & distribution(s) & dimensions variance &
correlation\tabularnewline
\midrule
\endhead
Uniform3 & 200*3 & Uniform & Equal & None\tabularnewline
         &       &(min = -0.5, max = 0.5)& &  \tabularnewline
Uniform15 & 200*15 & Uniform & Equal & None\tabularnewline
Uniform50 & 200*50 & Uniform & Equal & None\tabularnewline
Uniform150 & 200*150 & Uniform & Equal & None\tabularnewline
Uniform50c & 200*50 & Uniform & Equal & Random\tabularnewline
      & & & &  (between 0.1 and 0.9)\tabularnewline
Normal3 & 200*3 & Normal  & Equal & None\tabularnewline
          & & (mean = 0, sd = 1) & & \tabularnewline
Normal15 & 200*15 & Normal & Equal & None\tabularnewline
Normal50 & 200*50 & Normal & Equal & None\tabularnewline
Normal150 & 200*150 & Normal & Equal & None\tabularnewline
Normal50c & 200*50 & Normal & Equal & Random\tabularnewline
          & & & & (between 0.1 and 0.9)\tabularnewline
Random & 200*50 & Normal, Uniform, Lognormal & Equal & None\tabularnewline
 & &  (meanlog = 0, sdlog = 1) & &  \tabularnewline
PCA-like & 200*50 & Normal & Multiplicative & None\tabularnewline
PCO-like & 200*50 & Normal & Additive & None\tabularnewline
\bottomrule
\end{longtable}

Table 2: different simulated space distributions.

\renewcommand\baselinestretch{2}\selectfont

The differences in trait-space sizes (200 \(\times\) 3, 200 \(\times\)
15, 200 \(\times\) 50 or 200 \(\times\) 150) reflects the range of
dimensions in literature: ``low-dimension'' spaces (\(<15\)) are common
in ecology (Mammola 2019) whereas high dimension spaces (\(>100\)) are
common in macroevolution (Hopkins and Gerber 2017). We used a range of
distributions (uniform, normal or random) to test the effect of
observation distributions on the metrics. We used different levels of
variance for each dimensions in the spaces by making the variance on
each dimension either equal
(\(\sigma_{D1} \simeq \sigma_{D2} \simeq \sigma_{Di}\)) or decreasing
(\(\sigma_{D1} < \sigma_{D2} < \sigma_{Di}\)) with the decreasing factor
being either multiplicative (using the cumulative product of the inverse
of the number of dimensions: \(\prod_i^d(1/d)\)) or additive
(\(\sum_i^d(1/d)\)). Both multiplicative and cumulative reductions of
variance are used to illustrate the properties of ordinations where the
variance decreases per dimensions (healy2019; and in a normal way in
Multidimensional Scaling - MDS, PCO or PCoA; e.g. Close et al. 2015; in
a lognormal way in principal components analysis - PCA; e.g. Marcy et
al. 2016; Wright 2017). Finally, we added a correlation parameter to
take into account the potential correlation between different traits. We
repeated the simulation of each trait-space 20 times (resulting in 260
trait-spaces).

\subsection{Spatial occupancy metrics}\label{spatial-occupancy-metrics}

We then measured eight different metrics on the resulting transformed
spaces, including a new metric we produced, the average displacement,
which we expect to be influenced by changes in trait-space position.

\renewcommand\baselinestretch{1}\selectfont

\begin{longtable}[]{@{}lccll@{}}
\toprule
\begin{minipage}[b]{0.1\columnwidth}\raggedright\strut
Name\strut
\end{minipage} & \begin{minipage}[b]{0.25\columnwidth}\raggedright\strut
Definition\strut
\end{minipage} & \begin{minipage}[b]{0.09\columnwidth}\raggedright\strut
Captures\strut
\end{minipage} & \begin{minipage}[b]{0.1\columnwidth}\raggedright\strut
Source\strut
\end{minipage} & \begin{minipage}[b]{0.4\columnwidth}\raggedright\strut
Notes\strut
\end{minipage}\tabularnewline
\midrule
\endhead
\begin{minipage}[t]{0.1\columnwidth}\raggedright\strut
Average distance from centroid\strut
\end{minipage} & \begin{minipage}[t]{0.25\columnwidth}\raggedright\strut
\(\frac{\sqrt{\sum_{i}^{n}{({k}_{n}-Centroid_{k})^2}}}{d}\)\strut
\end{minipage} & \begin{minipage}[t]{0.09\columnwidth}\raggedright\strut
Volume\strut
\end{minipage} & \begin{minipage}[t]{0.1\columnwidth}\raggedright\strut
Lalibert√© and Legendre (2010)\strut
\end{minipage} & \begin{minipage}[t]{0.4\columnwidth}\raggedright\strut
equivalent to the functional dispersion (FDis) in Lalibert√© and Legendre
(2010) (without abundance)\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.1\columnwidth}\raggedright\strut
Sum of variances\strut
\end{minipage} & \begin{minipage}[t]{0.25\columnwidth}\raggedright\strut
\(\sum_{i}^{d}{\sigma^{2}{k_i}}\)\strut
\end{minipage} & \begin{minipage}[t]{0.09\columnwidth}\raggedright\strut
Volume\strut
\end{minipage} & \begin{minipage}[t]{0.1\columnwidth}\raggedright\strut
Wills (2001)\strut
\end{minipage} & \begin{minipage}[t]{0.4\columnwidth}\raggedright\strut
common metric used in palaeobiology (Ciampaglio et al. 2001)\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.1\columnwidth}\raggedright\strut
Sum of ranges\strut
\end{minipage} & \begin{minipage}[t]{0.25\columnwidth}\raggedright\strut
\(\sum_{i}^{d}{\|\text{max}(d_{i})-\text{min}(d_{i})\|}\)\strut
\end{minipage} & \begin{minipage}[t]{0.09\columnwidth}\raggedright\strut
Volume\strut
\end{minipage} & \begin{minipage}[t]{0.1\columnwidth}\raggedright\strut
Wills (2001)\strut
\end{minipage} & \begin{minipage}[t]{0.4\columnwidth}\raggedright\strut
more sensitive to outliers than the sum of variances\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.1\columnwidth}\raggedright\strut
Ellipsoid volume\strut
\end{minipage} & \begin{minipage}[t]{0.25\columnwidth}\raggedright\strut
\(\frac{\pi^{d/2}}{\Gamma(\frac{d}{2}+1)}\displaystyle\prod_{i}^{d} (\lambda_{i}^{0.5})\)\strut
\end{minipage} & \begin{minipage}[t]{0.09\columnwidth}\raggedright\strut
Volume\strut
\end{minipage} & \begin{minipage}[t]{0.1\columnwidth}\raggedright\strut
Donohue et al. (2013)\strut
\end{minipage} & \begin{minipage}[t]{0.4\columnwidth}\raggedright\strut
less sensitive to outliers than the convex hull hypervolume (D√≠az et al.
2016; Blonder 2018)\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.1\columnwidth}\raggedright\strut
Minimum spanning tree average distance\strut
\end{minipage} & \begin{minipage}[t]{0.25\columnwidth}\raggedright\strut
\(\frac{\sum(\text{branch length})}{n}\)\strut
\end{minipage} & \begin{minipage}[t]{0.09\columnwidth}\raggedright\strut
Density\strut
\end{minipage} & \begin{minipage}[t]{0.1\columnwidth}\raggedright\strut
Oksanen et al. (2007)\strut
\end{minipage} & \begin{minipage}[t]{0.4\columnwidth}\raggedright\strut
similar to the unscaled functional evenness (Vill√©ger et al. 2008)\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.1\columnwidth}\raggedright\strut
Minimum spanning tree distances evenness\strut
\end{minipage} & \begin{minipage}[t]{0.25\columnwidth}\raggedright\strut
\(\frac{\sum\text{min}\left(\frac{\text{branch length}}{\sum\text{branch length}}\right)-\frac{1}{n-1}}{1-\frac{1}{n-1}}\)\strut
\end{minipage} & \begin{minipage}[t]{0.09\columnwidth}\raggedright\strut
Density\strut
\end{minipage} & \begin{minipage}[t]{0.1\columnwidth}\raggedright\strut
Vill√©ger et al. (2008)\strut
\end{minipage} & \begin{minipage}[t]{0.4\columnwidth}\raggedright\strut
the functional evenness without weighted abundance (FEve; Vill√©ger et
al. 2008)\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.1\columnwidth}\raggedright\strut
Average nearest neighbour distance\strut
\end{minipage} & \begin{minipage}[t]{0.25\columnwidth}\raggedright\strut
\(min\left(\sqrt{\sum_{i}^{n}{({q}_{i}-p_{i})^2}}\right)\times \frac{1}{n}\)\strut
\end{minipage} & \begin{minipage}[t]{0.09\columnwidth}\raggedright\strut
Density\strut
\end{minipage} & \begin{minipage}[t]{0.1\columnwidth}\raggedright\strut
Foote (1990)\strut
\end{minipage} & \begin{minipage}[t]{0.4\columnwidth}\raggedright\strut
the density of pairs of observations in the trait-space\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.1\columnwidth}\raggedright\strut
Average displacement\strut
\end{minipage} & \begin{minipage}[t]{0.25\columnwidth}\raggedright\strut
\(\frac{\sqrt{\sum_{i}^{n}{({k}_{n})^2}}}{\sqrt{\sum_{i}^{n}{({k}_{n}-Centroid_{k})^2}}}\)\strut
\end{minipage} & \begin{minipage}[t]{0.09\columnwidth}\raggedright\strut
Position\strut
\end{minipage} & \begin{minipage}[t]{0.1\columnwidth}\raggedright\strut
This paper\strut
\end{minipage} & \begin{minipage}[t]{0.4\columnwidth}\raggedright\strut
the ratio between the observations' position from their centroid and the
centre of the trait-space. A value of 1 indicates that the observations'
centroid is the centre of the trait-space\strut
\end{minipage}\tabularnewline
\bottomrule
\end{longtable}

Table 3: List of metrics with \emph{n} being the number of observations,
\emph{d} the total number of dimensions, \emph{k} any specific row in
the matrix, \emph{Centroid} being their mean and \(\sigma^{2}\) their
variance. \(\Gamma\) is the Gamma distribution and \(\lambda_{i}\) the
eigen value of each dimension and \({q}_{i}\) and \(p_{i}\) are any
pairs of coordinates.

\renewcommand\baselinestretch{2}\selectfont


\subsection{Metric comparisons}\label{metric-comparisons}

We compared the space occupancy metrics correlations across all
simulations between each pair of metrics to assess they captured signal
(Vill√©ger et al. 2008; Lalibert√© and Legendre 2010). We used the metrics
on the full 13 trait-spaces described above. We then scaled the results
and measured the pairwise Pearson correlation to test whether metrics
were capturing a similar signal (high positive correlation), a different
signal (correlation close to 0) or an opposite signal (high negative
correlations) using the \texttt{psych} package (Revelle 2018).

\subsection{Changing space}\label{changing-spaces}

To measure how the metrics responded to changes within trait-spaces, we
removed 50\% of observations each time using the following algorithms:

\begin{itemize}
\item
  \textbf{Randomly:} by randomly removing 50\% of observations (Fig.
  2-A). This reflects a ``null'' biological model of changes in
  trait-space: the case when observations are removed regardless of
  their intrinsic characteristics. For example, if diversity is reduced
  by 50\% but the trait-space volume remains the same, there is a
  decoupling between diversity and space occupancy (Ruta et al. 2013).
  Our selected metrics are expected to not be affected by this change.
\item
  \textbf{Limit:} by removing all observations with a distance from the
  centre of the trait-space lower or greater than a radius \(\rho\)
  (where \(\rho\) is chosen such that 50\% observations are selected)
  generating two limit removals: \emph{maximum} and \emph{minimum}
  (respectively in orange and blue; Fig. 2-B). This can reflect a strict
  selection model where observations with trait values below or above a
  threshold are removed leading to an expansion or a contraction of the
  trait-space. Volume metrics are expected to be most affected by this
  change.
\item
  \textbf{Density:} by removing any pairs of point with a distance \(D\)
  from each other where (where \(D\) is chosen such that 50\%
  observations are selected) generating two density removals:
  \emph{high} and \emph{low} (respectively in orange and blue; Fig.
  2-C). This can reflect changes within groups in the trait-space due to
  ecological factors (e.g.~competition can generate niche repulsion -
  lower density; Grant and Grant 2006). Density metrics are expected to
  be most affected by this change.
\item
  \textbf{Position:} by removing points similarly as for \textbf{Limit}
  but using the distance from the furthest point from the centre
  generating two position removals: \emph{positive} and \emph{negative}
  (respectively in orange and blue; Fig. 2-D). This can reflect global
  changes in trait-space due, for example, to an entire group remaining
  diverse but occupying a different niche. Position metrics are expected
  to be most affected by this change.
\end{itemize}

The algorithm to select \(\rho\) or \(D\) is described in greater detail
in in the Supplementary material 1.

\begin{figure}
\centering
\includegraphics{shiftingspace_files/figure-latex/fig_reduce_space-1.pdf}
\caption{}
\end{figure}


To measure the effect of space reduction, distribution and
dimensionality on the metric, we scaled the metric to be relative to the
non-reduced space for each dimension distribution or number of
dimensions. We subtracted the observed occupancy with no space reduction
to all the occupancy measurements of the reduced spaces and then divided
it by the resulting maximum observed occupancy. Our occupancy metrics
where scaled between -1 and 1 with a value of 0 indicating no effect of
the space reduction and \(>0\) and \(<0\) respectively indicating an
increase or decrease in the occupancy metric value. We then measured the
probability of overlap of the between the non-random removals (limit,
density and position) and the random removals using the Bhattacharrya
Coefficient (probability of overlap between two distributions;
Bhattacharyya 1943).

\newpage
\renewcommand\baselinestretch{1}\selectfont

Figure 2: different type of space reduction. Each panel
displays two groups of 50\% of the data points each. Each group (orange
and blue) are generated using the following algorithm: A - randomly; B -
by limit (maximum and minimum limit); C - by density (high and low); and
D - by position (positive and negative). Panel E represents a typical
display of the reduction results displayed in Table 5: the dots
represent the median space occupancy values across all simulations for
each scenario of trait-space change (Table 2), the solid and dashed line
respectively the 50\% and 95\% confidence intervals. Results in grey are
the random 50\% reduction (panel A). Results in blue and orange
represent the opposite scenarios from panels B, C, and D. The displayed
value is the probability of overlap (Bhattacharrya Coefficient) between
the blue or orange distributions and the grey one.

\renewcommand\baselinestretch{2}\selectfont

\subsubsection{Measuring the effect of space and dimensionality on
shifting
spaces}\label{measuring-the-effect-of-space-and-dimensionality-on-shifting-spaces}

Distribution differences and the number of dimensions can have an effect
on the metric results. For example, in a normally distributed space, a
decrease in density can often lead to an increase in volume. This is not
necessarily true in log-normal spaces or in uniform spaces for certain
metrics. Furthermore, high dimensional spaces (\textgreater{}10) are
subject to the ``curse of multidimensionality'' (Ch√°vez et al. 2001):
data becomes sparser with increasing number of dimensions, such that the
probability of two points A and B overlapping in \emph{n} dimensions is
the product of the probability of the two points overlapping on each
dimensions(\(\prod_{i}^{d} P(A = B)_{Di}\)). This probability decreases
as a product of the number of dimensions. Therefore, the ``curse'' can
make the interpretation of high dimensional data counter-intuitive. For
example if a group expands in multiple dimensions (i.e.~increase in
volume), the actual hypervolume can decrease (Fig. 3 and Tables 6, 7).

We measured the effect of space distribution and dimensionality using an
ANOVA (\(occupancy \sim distribution\) and
\(occupancy \sim dimensions\)) by using all spaces with 50 dimensions
and the uniform and normal spaces with equal variance and no correlation
with 3, 15, 50, 100 and 150 dimensions (Table 2) for testing
respectively the effect of distribution and dimensions. The results of
the ANOVAs (\emph{p}-values) are reported in Table 5 (see supplementary
material 1 for the full ANOVA result tables).

\subsection{Empirical examples}\label{empirical-examples}

We analysed the effect of the different space occupancy metrics on six
different empirical studies covering a broad range of fields that employ
trait-space analyses (palaeobiology, macroevolution, evo-devo, ecology,
etc.). For each of these studies we generated trait-spaces from the data
published with the papers. We divided each trait-spaces into two
biologically-relevant groups and tested whether the metrics
differentiated the groups in different ways. Both the grouping and the
questions where based on a simplified version of the topics of these
papers (with no intention to re-analyse the data but to be
representative of the diversity of questions in ecology and evolution).
The procedures to generate the data and the groups varies from one study
to the other but is detailed and fully reproducible in the supplementary
materials 2.

\renewcommand\baselinestretch{1}\selectfont


\begin{longtable}[]{@{}llllllll@{}}
\toprule
\begin{minipage}[b]{0.06\columnwidth}\raggedright\strut
study\strut
\end{minipage} & \begin{minipage}[b]{0.08\columnwidth}\raggedright\strut
field\strut
\end{minipage} & \begin{minipage}[b]{0.1\columnwidth}\raggedright\strut
taxonomic Group\strut
\end{minipage} & \begin{minipage}[b]{0.13\columnwidth}\raggedright\strut
traits (data)\strut
\end{minipage} & \begin{minipage}[b]{0.11\columnwidth}\raggedright\strut
trait-space\strut
\end{minipage} & \begin{minipage}[b]{0.06\columnwidth}\raggedright\strut
size\strut
\end{minipage} & \begin{minipage}[b]{0.07\columnwidth}\raggedright\strut
groups (orange/blue in Table 6)\strut
\end{minipage} & \begin{minipage}[b]{0.15\columnwidth}\raggedright\strut
type of question\strut
\end{minipage}\tabularnewline
\midrule
\endhead
\begin{minipage}[t]{0.06\columnwidth}\raggedright\strut
Beck and Lee (2014)\strut
\end{minipage} & \begin{minipage}[t]{0.08\columnwidth}\raggedright\strut
Palaeontology\strut
\end{minipage} & \begin{minipage}[t]{0.1\columnwidth}\raggedright\strut
Mammalia\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
discrete morphological phylogenetic data\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
Ordination of a distance matrix (PCO)\strut
\end{minipage} & \begin{minipage}[t]{0.06\columnwidth}\raggedright\strut
106*105\strut
\end{minipage} & \begin{minipage}[t]{0.07\columnwidth}\raggedright\strut
52 crown vs.~54 stem\strut
\end{minipage} & \begin{minipage}[t]{0.15\columnwidth}\raggedright\strut
Are crown mammals more disparate than stem mammals?\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.06\columnwidth}\raggedright\strut
Wright (2017)\strut
\end{minipage} & \begin{minipage}[t]{0.08\columnwidth}\raggedright\strut
Palaeontology\strut
\end{minipage} & \begin{minipage}[t]{0.1\columnwidth}\raggedright\strut
Crinoidea\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
discrete morphological phylogenetic data\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
Ordination of a distance matrix (PCO)\strut
\end{minipage} & \begin{minipage}[t]{0.06\columnwidth}\raggedright\strut
42*41\strut
\end{minipage} & \begin{minipage}[t]{0.07\columnwidth}\raggedright\strut
16 before vs.~23 after\strut
\end{minipage} & \begin{minipage}[t]{0.15\columnwidth}\raggedright\strut
Is there a difference in disparity before and after the Ordovician mass
extinction?\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.06\columnwidth}\raggedright\strut
Marcy et al. (2016)\strut
\end{minipage} & \begin{minipage}[t]{0.08\columnwidth}\raggedright\strut
Evolution\strut
\end{minipage} & \begin{minipage}[t]{0.1\columnwidth}\raggedright\strut
Rodentia\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
skull 2D landmark coordinates\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
Ordination of a Procrustes Superimposition (PCA)\strut
\end{minipage} & \begin{minipage}[t]{0.06\columnwidth}\raggedright\strut
454*134\strut
\end{minipage} & \begin{minipage}[t]{0.07\columnwidth}\raggedright\strut
225 \emph{Megascapheus} vs.~229 \emph{Thomomys}\strut
\end{minipage} & \begin{minipage}[t]{0.15\columnwidth}\raggedright\strut
Are two genera of gopher morphologically distinct?\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.06\columnwidth}\raggedright\strut
Hopkins and Pearson (2016)\strut
\end{minipage} & \begin{minipage}[t]{0.08\columnwidth}\raggedright\strut
Evolution\strut
\end{minipage} & \begin{minipage}[t]{0.1\columnwidth}\raggedright\strut
Trilobita\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
3D landmark coordinates\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
Ordination of a Procrustes Superimposition (PCA)\strut
\end{minipage} & \begin{minipage}[t]{0.06\columnwidth}\raggedright\strut
46*46\strut
\end{minipage} & \begin{minipage}[t]{0.07\columnwidth}\raggedright\strut
36 adults vs.~10 adults\strut
\end{minipage} & \begin{minipage}[t]{0.15\columnwidth}\raggedright\strut
Are juvenile trilobites a subset of adult ones?\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.06\columnwidth}\raggedright\strut
Jones et al. (2015)\strut
\end{minipage} & \begin{minipage}[t]{0.08\columnwidth}\raggedright\strut
Ecology\strut
\end{minipage} & \begin{minipage}[t]{0.1\columnwidth}\raggedright\strut
Plantae\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
Communities species compositions\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
Ordination of a Jaccard distance matrix (PCO)\strut
\end{minipage} & \begin{minipage}[t]{0.06\columnwidth}\raggedright\strut
48*47\strut
\end{minipage} & \begin{minipage}[t]{0.07\columnwidth}\raggedright\strut
24 aspens vs.~24 grasslands\strut
\end{minipage} & \begin{minipage}[t]{0.15\columnwidth}\raggedright\strut
Is there a difference in species composition between aspens and
grasslands?\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.06\columnwidth}\raggedright\strut
Healy et al. (2019)\strut
\end{minipage} & \begin{minipage}[t]{0.08\columnwidth}\raggedright\strut
Ecology\strut
\end{minipage} & \begin{minipage}[t]{0.1\columnwidth}\raggedright\strut
Animalia\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
Life history traits\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
Ordination of continuous traits (PCA)\strut
\end{minipage} & \begin{minipage}[t]{0.06\columnwidth}\raggedright\strut
285*6\strut
\end{minipage} & \begin{minipage}[t]{0.07\columnwidth}\raggedright\strut
83 ecthotherms vs.~202 endotherms\strut
\end{minipage} & \begin{minipage}[t]{0.15\columnwidth}\raggedright\strut
Do endotherms have more diversified life history strategies than
ectotherms?\strut
\end{minipage}\tabularnewline
\bottomrule
\end{longtable}

\renewcommand\baselinestretch{2}\selectfont

Table 4: details of the six empirical trait-spaces.

For each empirical trait-space we bootstrapped each group 500 times
(Guillerme 2018) and applied the eight space occupancy metric to each
pairs of groups. We then compared the means of each groups using the
Bhattacharrya Coefficient (Bhattacharyya 1943).

\section{Results}\label{results}

\subsection{Metric comparisons}\label{metric-comparisons-1}

\begin{figure}
\centering
\includegraphics{shiftingspace_files/figure-latex/fig_metric_correlation-1.pdf}
\caption{pairwise correlation between the scaled metrics.
Numbers on the upper right corner are the Pearson correlations. The red
line are linear regressions (with the confidence intervals in grey).}
\end{figure}

All the metrics were either positively correlated (Pearson correlation
of 0.99 for the average distance from centroid and sum of variance or
0.97 for the average nearest neighbour distance and minimum spanning
tree average length; Fig. 3) or somewhat correlated (ranging from 0.66
for the sum of variances and the ellipsoid volume to -0.09 between the
average displacement and the average distance from centroid; Fig. 3).
All metrics but the ellipsoid volume were normally (or nearly normally)
distributed (Fig. 3). More comparisons between metrics are available in
the supplementary materials 3.

\subsection{Space shifting}\label{space-shifting}

\renewcommand\baselinestretch{1}\selectfont

\begin{longtable}[]{@{}llllll@{}}
\toprule
\begin{minipage}[b]{0.10\columnwidth}\raggedright\strut
Metric\strut
\end{minipage} & \begin{minipage}[b]{0.13\columnwidth}\raggedright\strut
Volume change\strut
\end{minipage} & \begin{minipage}[b]{0.14\columnwidth}\raggedright\strut
Density change\strut
\end{minipage} & \begin{minipage}[b]{0.13\columnwidth}\raggedright\strut
Position change\strut
\end{minipage} & \begin{minipage}[b]{0.17\columnwidth}\raggedright\strut
Distribution effect\strut
\end{minipage} & \begin{minipage}[b]{0.16\columnwidth}\raggedright\strut
Dimensions effect\strut
\end{minipage}\tabularnewline
\midrule
\endhead
\begin{minipage}[t]{0.10\columnwidth}\raggedright\strut
Average distance from centroid\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-1.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.14\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-2.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-3.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright\strut
p = 0.449\strut
\end{minipage} & \begin{minipage}[t]{0.16\columnwidth}\raggedright\strut
p = 0.958\strut
\end{minipage}\tabularnewline
\hline
\begin{minipage}[t]{0.10\columnwidth}\raggedright\strut
Sum of variances\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-4.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.14\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-5.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-6.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright\strut
p = 0.274\strut
\end{minipage} & \begin{minipage}[t]{0.16\columnwidth}\raggedright\strut
p = 0.873\strut
\end{minipage}\tabularnewline
\hline
\begin{minipage}[t]{0.10\columnwidth}\raggedright\strut
Sum of ranges\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-7.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.14\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-8.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-9.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright\strut
p = 0 ***\strut
\end{minipage} & \begin{minipage}[t]{0.16\columnwidth}\raggedright\strut
p = 0 ***\strut
\end{minipage}\tabularnewline
\hline
\begin{minipage}[t]{0.10\columnwidth}\raggedright\strut
Ellipsoid volume\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-10.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.14\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-11.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-12.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright\strut
p = 0 ***\strut
\end{minipage} & \begin{minipage}[t]{0.16\columnwidth}\raggedright\strut
p = 0 ***\strut
\end{minipage}\tabularnewline
\hline
\begin{minipage}[t]{0.10\columnwidth}\raggedright\strut
Minimum spanning tree average distance\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-13.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.14\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-14.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-15.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright\strut
p = 0.326\strut
\end{minipage} & \begin{minipage}[t]{0.16\columnwidth}\raggedright\strut
p = 0.435\strut
\end{minipage}\tabularnewline
\hline
\begin{minipage}[t]{0.10\columnwidth}\raggedright\strut
Minimum spanning tree distances evenness\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-16.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.14\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-17.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-18.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright\strut
p = 0 ***\strut
\end{minipage} & \begin{minipage}[t]{0.16\columnwidth}\raggedright\strut
p = 0 ***\strut
\end{minipage}\tabularnewline
\hline
\begin{minipage}[t]{0.10\columnwidth}\raggedright\strut
Average nearest neighbour distance\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-19.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.14\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-20.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-21.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright\strut
p = 0.207\strut
\end{minipage} & \begin{minipage}[t]{0.16\columnwidth}\raggedright\strut
p = 0.626\strut
\end{minipage}\tabularnewline
\hline
\begin{minipage}[t]{0.10\columnwidth}\raggedright\strut
Average displacements\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-22.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.14\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-23.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results-24.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright\strut
p = 0 ***\strut
\end{minipage} & \begin{minipage}[t]{0.16\columnwidth}\raggedright\strut
p = 0 ***\strut
\end{minipage}\tabularnewline
\bottomrule
\end{longtable}


Table 5: Results of the effect of space reduction, space dimension
distributions and dimensions number of the different space occupancy
metrics. See Fig. 2 for interpretation of the figures. \emph{p}-values
for distribution effect and dimensions effect represents respectively
the effect of the ANOVAs space occupancy \textasciitilde{} distributions
and space occupancy \textasciitilde{} dimensions (0 `***' 0.001 `**'
0.01 `*' 0.05 `.' 0.1 '' 1).

\renewcommand\baselinestretch{2}\selectfont


As expected, some different metrics capture different aspects of space
occupancy. However, it can be hard to predict the behaviour of each
metric when 50\% of the observations are removed. We observe a clear
decrease in median metric in less than a third of the space reductions
(10/36).

In terms of change in volume, only the average distance from centroid
and the sum of variances seem to capture a clear change in both
directions. However, the increase in volume does not correspond to an
\emph{actual} increase in volume in the trait-space (i.e.~the volume
from the blue observations in Fig. 2-B is equivalent to the one in Fig.
2-A). In terms of change in density, only the minimum spanning tree
average distance and the average nearest neighbour distance seem to
capture a clear change in both directions. In terms of change in
position, only the average displacement metric seems to capture a change
a clear change in direction (albeit not in both directions). This is not
surprising, since the notion of positions becomes more and more complex
to appreciate as dimensionality increases (i.e.~beyond left/right,
up/down and front/back).

\subsection{Empirical example}\label{empirical-example}

\renewcommand\baselinestretch{1}\selectfont


\begin{longtable}[]{@{}lllllll@{}}
\toprule
\begin{minipage}[b]{0.09\columnwidth}\raggedright\strut
Metric\strut
\end{minipage} & \begin{minipage}[b]{0.11\columnwidth}\raggedright\strut
Beck and Lee 2014\strut
\end{minipage} & \begin{minipage}[b]{0.12\columnwidth}\raggedright\strut
Wright 2017\strut
\end{minipage} & \begin{minipage}[b]{0.13\columnwidth}\raggedright\strut
Marcy et al. 2016\strut
\end{minipage} & \begin{minipage}[b]{0.11\columnwidth}\raggedright\strut
Hopkins and Pearson 2016\strut
\end{minipage} & \begin{minipage}[b]{0.13\columnwidth}\raggedright\strut
Jones et al. 2015\strut
\end{minipage} & \begin{minipage}[b]{0.11\columnwidth}\raggedright\strut
Healy et al. 2019\strut
\end{minipage}\tabularnewline
\hline
\midrule
\endhead
\begin{minipage}[t]{0.09\columnwidth}\raggedright\strut
Average distance from centroid\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-1.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-9.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-17.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-25.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-33.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-41.pdf}\strut
\end{minipage}\tabularnewline
\hline
\begin{minipage}[t]{0.09\columnwidth}\raggedright\strut
Sum of variances\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-2.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-10.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-18.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-26.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-34.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-42.pdf}\strut
\end{minipage}\tabularnewline
\hline
\begin{minipage}[t]{0.09\columnwidth}\raggedright\strut
Sum of ranges\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-3.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-11.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-19.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-27.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-35.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-43.pdf}\strut
\end{minipage}\tabularnewline
\hline
\begin{minipage}[t]{0.09\columnwidth}\raggedright\strut
Ellipsoid volume\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-4.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-12.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-20.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-28.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-36.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-44.pdf}\strut
\end{minipage}\tabularnewline
\hline
\begin{minipage}[t]{0.09\columnwidth}\raggedright\strut
Minimum spanning tree average distance\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-5.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-13.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-21.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-29.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-37.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-45.pdf}\strut
\end{minipage}\tabularnewline
\hline
\begin{minipage}[t]{0.09\columnwidth}\raggedright\strut
Minimum spanning tree distances evenness\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-6.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-14.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-22.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-30.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-38.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-46.pdf}\strut
\end{minipage}\tabularnewline
\hline
\begin{minipage}[t]{0.09\columnwidth}\raggedright\strut
Average nearest neighbour distance\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-7.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-15.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-23.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-31.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-39.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-47.pdf}\strut
\end{minipage}\tabularnewline
\hline
\begin{minipage}[t]{0.09\columnwidth}\raggedright\strut
Average displacements\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-8.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.12\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-16.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-24.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-32.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.13\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-40.pdf}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright\strut
\includegraphics{shiftingspace_files/figure-latex/fable_results_empirical-48.pdf}\strut
\end{minipage}\tabularnewline
\hline
\bottomrule
\end{longtable}

Table 6: Comparisons of pairs of groups in different empirical
trait-spaces. NAs are used for cases where space occupancy could not be
measured due to the curse of multidimensionality. The displayed values
are the probability of overlap between both groups (Bhattacharrya
Coefficient).

\renewcommand\baselinestretch{2}\selectfont


Similarly as for the simulated results, the empirical ones indicate that
there is no perfect one-size-fit all metric. For all eight metrics
(expect the ellipsoid volume) we see either one group or the other
having a bigger mean than the other and no consistent case where a group
has a bigger mean than the other for all the metrics. For example, in
the Beck and Lee (2014)'s dataset, there is a clear non-overlap in space
occupancy volume using the average distance from centroid or the sum of
variances (overlaps of respectively 0.175 and 0.159) but no overlap when
measuring the volume using the sum of ranges (0.966). However, for the
Hopkins and Pearson (2016)'s dataset, this pattern is reversed (no clear
differences for the average distance from centroid or the sum of
variances - 0.701 and 0.865 respectively) but a clear difference for the
sum of ranges (0). Furthermore, for each dataset, the absolute
differences between each groups is not consistent depending on the
metrics. For example, in Hopkins and Pearson (2016)'s dataset, the
orange group's mean is clearly higher than the blue one when measuring
the sum of ranges (0) and the inverse is true when measuring the average
displacement (0).

\section{Discussion}\label{discussion}

Here we tested 25 metrics of trait-space occupancy on simulated and
empirical datasets to assess how each metric captures changes in
trait-space volume, density and position. Our results show that the
correlation between metrics can vary both within and between metric
categories (Fig. 3), highlighting the importance of understanding the
metric classification for the interpretation of results. Furthermore,
our simulations show that different metrics capture different types of
trait-space change (Table 5), meaning that the use of multiple metrics
is important for comprehensive interpretation of trait-space occupancy.
We also show that the choice of metric impacts the interpretation of
group differences in empirical datasets (Table 6), again emphasizing
that metric choice has a real impact on the interpretation of specific
biological questions

\paragraph{Metrics comparisons}\label{metrics-comparisons}

Metrics within the same category of trait-space occupancy (volume,
density or position) do not have the same level of correlation with each
other. For example, the average distance from centroid (volume) is
highly correlated to the sum of variances (volume - correlation of 0.99)
and somewhat correlated with the minimum spanning tree average distance
(density - correlation of 0.66) but poorly with the ellipsoid volume
(volume - correlation of 0.17) and the minimum spanning tree distances
evenness (density - correlation of -0.05). Furthermore, the fact that we
have such a range of correlations for normal distributions suggests that
each metric can capture different summaries of space occupancy ranging
from obvious differences (for metrics not strongly correlated) to subtle
ones (for metrics strongly correlated).

\paragraph{Space shifting}\label{space-shifting-1}

Most metrics capture no changes in space occupancy for the ``null''
(random) space reduction (in grey in Table 5). This is a desirable
behaviour for space occupancy metrics since it will likely avoid false
positive errors in empirical studies that estimate biological processes
from space occupancy patterns (e.g.~competition Brusatte et al. 2008,
convergence Marcy et al. (2016), life history traits Healy et al.
(2019)). However, the average nearest neighbour distance and the sum of
ranges have a respectively positive and negative ``null'' median. This
is not especially a bad property but it should be kept in mind that even
random processes can increase or decrease these metric value.

Regarding the changes in volume, the sum of variances and the average
distance from centroid are good descriptors (Table 5). However, as
illustrated in the 2D examples in Fig. 2-B only the blue change results
(maximum limit - Table 5) should not result in a direct change in volume
since the trait-space is merely ``hollowed'' out. That said,
``hollowing'' is more hard to conceptualise in many dimensions and the
metrics can still be interpreted for comparing groups (orange has a
smaller volume than blue).

Regarding changes in density, the average nearest neigbhour distance and
the minimum spanning tree average distance consistently detect changes
in density with more precision for low density trait-spaces (in blue in
Table 5). However, we can observe some degree of correlation between the
changes in density and the changes in volume for most metric picking
either signal. This could be due to the use of normally distributed
spaces where a change in density often leads to a change in volume. This
is not necessary the case with empirical data.

Regarding the changes in position of the trait-space, all but the
average displacement metric seems to not be able to distinguish between
a random change and a displacement of the trait-space (Table 5).
Furthermore, the average displacement metric does not distinguish
between and positive or a negative displacement of the trait-space: this
might be due to the inherent complexity of \emph{position} in a
multidimensional trait-space.

\paragraph{Empirical examples}\label{empirical-examples-1}

Although most differences are fairly consistent within each dataset with
one group having a higher space occupancy score than the other for
multiple metrics, this difference can be more or less pronounced within
each dataset (ranging from no to nearly full overlap - BC
\(\in(0;0.995)\)) and sometimes even reversed. This indicates that
opposite conclusions can be drawn from a dataset depending on which
space occupancy metric is considered. For example, in Wright (2017),
crinoids after the Ordovician mass extinction have a higher median
metric value for all metrics but for the average displacement. These
differences depending on the metrics are also more pronounced in the
empirical datasets where the observations per group are unequal (Hopkins
and Pearson 2016; Healy et al. 2019). \#\#\# Caveats

While our simulations have been useful to illustrate the behavior of
diverse space occupancy metrics, they have several caveats. First, the
simulated observations in the trait-spaces are independent. This is not
the case in biology where observations can be spatially (Jones et al.
2015) or phylogenetically correlated (e.g. Beck and Lee 2014). Second,
the algorithm used to reduce the trait-spaces might not always
accurately empirical changes. This might favour some specific metrics
over others, in particular for the changes in density that modifies the
nearest neighbour density rather than changing the global density. This
algorithmic choice was made in order to not confound changes in density
along with changes in volume. However, the results presented here
probably capture the general behaviour of each metric since results are
consistent between the simulated and empirical analysis. Furthermore,
\href{https://tguillerme.shinyapps.io/moms/}{\texttt{moms}} allows to
test the caveats mentioned above by uploading empirical trait-space.

\subsubsection{Suggestions}\label{suggestions}

We insist that no metric is better than the next one and that
researchers should use the most appropriate metrics based on the metric
and trait-space properties as well as their specific biological
question. However, following the findings of this study we suggest
several points:

First, we suggest using multiple metrics to tackle different aspects of
the trait-space. This follows the same logical thinking that the mean
might not be sufficient to describe a distribution (e.g.~the variance
might be good additional descriptor). Although using multiple metrics is
not uncommon in macroevolutionary studies (e.g. Halliday and Goswami
2016) or in ecology (Mammola 2019), they often do not cover contrasted
aspects of the trait-space.

Second, we suggest selecting the metrics that best help answering the
biological question a hand. If one studies an adaptive radiation in a
group of organisms, it is worth thinking what would be the expected null
model: would the group's volume increase (radiation in all directions),
would it increase in density (niche specialisation) or would it shift in
position (radiation into a new set of niches)?

Third, we suggest to not name metrics as the biological aspect they are
describing (e.g. ``disparity'' or ``functional dispersion'') but rather
what they are measuring (e.g. ``sum of dimensions variance''). We
believe this will allow both a clearer understanding of what \emph{is}
measured and a better communication between ecology and evolution
research where metrics can be similar but have different names (Fig. 3).

Multidimensional analyses have been acknowledged to be an essential
tool-kit modern biology but can often be counter-intuitive (Ch√°vez et
al. 2001). It is thus crucial to accurately describe patterns in
multidimensional trait-spaces to be able to link them to biological
processes. When summarising trait-spaces, it is important to remember
that a pattern captured by a specific space occupancy metric is often
dependent on the properties of the trait-space and of the particular
biological question of interest. We believe that having a clearer
understanding of both the properties of the trait-space and the
associated space occupancy metrics (e.g.~using
\href{https://tguillerme.shinyapps.io/moms/}{\texttt{moms}}) as well as
using novel space occupancy metrics to answer specific questions will be
of great use to study biological processes in a multidimensional world.

\section{Acknowledgements}\label{acknowledgements}

We thank Natalie Jones and Kevin Healy for helping with the empirical
ecological datasets. We acknowledge funding from the Australian Research
Council DP170103227 and FT180100634 awarded to VW.

\section{Authors contributions}\label{authors-contributions}

TG, MNP, AEM and VW designed the project. TG and AEM collected the
empirical dataset. TG ran the analyses and designed the software. TG,
MNP, AEM and VW wrote the manuscript.

\section{Data Availability, repeatability and
reproducibility}\label{data-availability-repeatability-and-reproducibility}

The raw empirical data is available from the original papers (Beck and
Lee 2014; Jones et al. 2015, Marcy et al. (2016); Hopkins and Pearson
2016; Wright 2017; Healy et al. 2019). The subsets of the empirical data
used in this analysis are available on figshare
\href{https://doi.org/10.6084/m9.figshare.9943181.v1}{DOI:
10.6084/m9.figshare.9943181.v1}. The modified empirical data are
available in the package accompanying this manuscript
(\texttt{data(moms::demo\_data)}). This manuscript (including the
figures, tables and supplementary material) is repeatable and
reproducible by compiling the vignette of the
\href{https://github/TGuillerme/moms}{GitHub \texttt{moms\ R} package}.
The code for the \texttt{moms} shiny app is available from the
\href{https://github/TGuillerme/moms}{GitHub \texttt{moms\ R} package}.

\section*{References}\label{references}
\addcontentsline{toc}{section}{References}

\hypertarget{refs}{}
\hypertarget{ref-adams2013geomorph}{}
Adams D.C., Ot√°rola-Castillo E. 2013. Geomorph: An R package for the
collection and analysis of geometric morphometric shape data. Methods in
Ecology and Evolution. 4:393--399.

\hypertarget{ref-beck2014}{}
Beck R.M.D., Lee M.S.Y. 2014. Ancient dates or accelerated rates?
Morphological clocks and the antiquity of placental mammals. Proceedings
of the Royal Society B: Biological Sciences. 281:20141278.

\hypertarget{ref-bhattacharyya1943}{}
Bhattacharyya A. 1943. On a measure of divergence between two
statistical populations defined by their probability distributions.
Bulletin of the Calcutta Mathematical Society. 35:99--109.

\hypertarget{ref-blonder2018}{}
Blonder B. 2018. Hypervolume concepts in niche-and trait-based ecology.
Ecography. 41:1441--1455.

\hypertarget{ref-momocs}{}
Bonhomme V., Picq S., Gaucherel C., Claude J. 2014. Momocs: Outline
analysis using R. Journal of Statistical Software. 56:1--24.

\hypertarget{ref-brusatte2008}{}
Brusatte S.L., Benton M.J., Ruta M., Lloyd G.T. 2008. Superiority,
competition, and opportunism in the evolutionary radiation of dinosaurs.
Science. 321:1485--1488.

\hypertarget{ref-cursedimensionality}{}
Ch√°vez E., Navarro G., Baeza-Yates R., Marroqu√≠n J.L. 2001. Searching in
metric spaces. ACM Comput. Surv. 33:273--321.

\hypertarget{ref-ciampaglio2001}{}
Ciampaglio C.N., Kemp M., McShea D.W. 2001. Detecting changes in
morphospace occupation patterns in the fossil record: Characterization
and analysis of measures of disparity. Paleobiology. 71:695--715.

\hypertarget{ref-close2015}{}
Close R.A., Friedman M., Lloyd G.T., Benson R.B. 2015. Evidence for a
mid-Jurassic adaptive radiation in mammals. Current Biology.

\hypertarget{ref-diaz2016}{}
D√≠az S., Kattge J., Cornelissen J.H., Wright I.J., Lavorel S., Dray S.,
Reu B., Kleyer M., Wirth C., Prentice I.C., others. 2016. The global
spectrum of plant form and function. Nature. 529:167.

\hypertarget{ref-donohue2013}{}
Donohue I., Petchey O.L., Montoya J.M., Jackson A.L., McNally L., Viana
M., Healy K., Lurgi M., O'Connor N.E., Emmerson M.C. 2013. On the
dimensionality of ecological stability. Ecology Letters. 16:421--429.

\hypertarget{ref-endler2005}{}
Endler J.A., Westcott D.A., Madden J.R., Robson T. 2005. Animal visual
systems and the evolution of color patterns: Sensory processing
illuminates signal evolution. Evolution. 59:1795--1818.

\hypertarget{ref-foote1990}{}
Foote M. 1990. Nearest-neighbor analysis of trilobite morphospace.
Systematic Zoology. 39:371--382.

\hypertarget{ref-grant2006}{}
Grant P.R., Grant B.R. 2006. Evolution of character displacement in
darwins finches. Science. 313:224--226.

\hypertarget{ref-disprity}{}
Guillerme T. 2018. dispRity: A modular R package for measuring
disparity. Methods in Ecology and Evolution. 9:1755--1763.

\hypertarget{ref-halliday2015}{}
Halliday T.J.D., Goswami A. 2016. Eutherian morphological disparity
across the end-cretaceous mass extinction. Biological Journal of the
Linnean Society. 118:152--168.

\hypertarget{ref-geiger2008}{}
Harmon L.J., Weir J.T., Brock C.D., Glor R.E., Challenger W. 2008.
GEIGER: Investigating evolutionary radiations. Bioinformatics.
24:129--131.

\hypertarget{ref-healy2019}{}
Healy K., Ezard T.H.G., Jones O.R., Salguero-G'omez R., Buckley Y.M.
2019. Animal life history is shaped by the pace of life and the
distribution of age-specific mortality and reproduction. Nature Ecology
\& Evolution. 2397-334X.

\hypertarget{ref-hopkins2016}{}
Hopkins M., Pearson K. 2016. Non-linear ontogenetic shape change in
cryptolithus tesselatus (trilobita) using three-dimensional geometric
morphometrics. Palaeontologia Electronica. 19:1--54.

\hypertarget{ref-hopkins2017}{}
Hopkins M.J., Gerber S. 2017. Morphological disparity. In: Nuno de la
Rosa L., M√ºller G., editors. Evolutionary developmental biology: A
reference guide. Cham: Springer International Publishing. p. 1--12.

\hypertarget{ref-jones2015}{}
Jones N.T., Germain R.M., Grainger T.N., Hall A.M., Baldwin L., Gilbert
B. 2015. Dispersal mode mediates the effect of patch size and patch
connectivity on metacommunity diversity. Journal of Ecology.
103:935--944.

\hypertarget{ref-lalibertuxe92010}{}
Lalibert√© √â., Legendre P. 2010. A distance-based framework for measuring
functional diversity from multiple traits. Ecology. 91:299--305.

\hypertarget{ref-legendre2012}{}
Legendre P., Legendre L.F. 2012. Numerical ecology. Elsevier.

\hypertarget{ref-mammola2019}{}
Mammola S. 2019. Assessing similarity of n-dimensional hypervolumes:
Which metric to use? Journal of Biogeography. 0.

\hypertarget{ref-marcy2016}{}
Marcy A.E., Hadly E.A., Sherratt E., Garland K., Weisbecker V. 2016.
Getting a head in hard soils: Convergent skull evolution and divergent
allometric patterns explain shape variation in a highly diverse genus of
pocket gophers (thomomys). BMC evolutionary biology. 16:207.

\hypertarget{ref-oksanen2007vegan}{}
Oksanen J., Kindt R., Legendre P., O'Hara B., Stevens M.H.H., Oksanen
M.J., Suggests M. 2007. The vegan package. Community ecology package.
10:631--637.

\hypertarget{ref-psych}{}
Revelle W. 2018. Psych: Procedures for psychological, psychometric, and
personality research. Evanston, Illinois: Northwestern University.

\hypertarget{ref-ruta2013}{}
Ruta M., Angielczyk K.D., Fr√∂bisch J., Benton M.J. 2013. Decoupling of
morphological disparity and taxic diversity during the adaptive
radiation of anomodont therapsids. Proceedings of the Royal Society of
London B: Biological Sciences. 280.

\hypertarget{ref-villuxe9ger2008}{}
Vill√©ger S., Mason N.W.H., Mouillot D. 2008. New multidimensional
functional diversity indices for a multifaceted framework in functional
ecology. Ecology. 89:2290--2301.

\hypertarget{ref-wills2001}{}
Wills M.A. 2001. Morphological disparity: A primer. In: Adrain J.M.,
Edgecombe G.D., Lieberman B.S., editors. Fossils, phylogeny, and form.
Springer US. p. 55--144.

\hypertarget{ref-wright2017}{}
Wright D.F. 2017. Phenotypic innovation and adaptive constraints in the
evolutionary radiation of palaeozoic crinoids. Scientific Reports.
7:13745.


\end{document}
